{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Version History\n",
    "#print(\"Version 1.0.0: 09/15/2022 2:06pm - Nate Calvanese - First version created\")\n",
    "#print(\"Version 1.0.1: 09/16/2022 3:10pm - Nate Calvanese - Shifted from transform to mapping compatibility\")\n",
    "#print(\"Version 1.0.2: 10/14/2022 7:40pm - Nate Calvanese - Added compatibility evaluation and support for multiple mapping specs\")\n",
    "#print(\"Version 1.0.3: 10/18/2022 1:33pm - Nate Calvanese - Encoded column names to match mapping specifications\")\n",
    "print(\"Version 1.0.4: 10/20/2022 11:50am - Nate Calvanese - Added ability to pull schemas for workspaces missing from workspace_schemas.csv\")\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#!pip install --upgrade import_ipynb"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Main Script"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "workspace name = anvil_workspace_ingest_resources_dev\n",
      "workspace project = dsp-data-ingest\n",
      "workspace bucket = gs://fc-2a9eefc3-0302-427f-9ac3-82f078741c03\n",
      "workspace bucket name = fc-2a9eefc3-0302-427f-9ac3-82f078741c03\n"
     ]
    }
   ],
   "source": [
    "## Imports and environment variables\n",
    "# imports\n",
    "import import_ipynb\n",
    "import pandas as pd\n",
    "import json\n",
    "import re\n",
    "import os\n",
    "from google.cloud import storage\n",
    "from firecloud import api as fapi\n",
    "import ingest_pipeline_utilities as utils\n",
    "import build_mapping_query as bmq\n",
    "\n",
    "# Configure pandas display\n",
    "pd.set_option('display.max_rows', None)\n",
    "pd.set_option('display.max_columns', None)\n",
    "pd.set_option(\"display.max_colwidth\", None)\n",
    "pd.set_option('display.width', 1000)\n",
    "pd.set_option('display.colheader_justify', 'center')\n",
    "pd.set_option('display.precision', 3)\n",
    "\n",
    "# workspace environment variables\n",
    "ws_name = os.environ[\"WORKSPACE_NAME\"]\n",
    "ws_project = os.environ[\"WORKSPACE_NAMESPACE\"]\n",
    "ws_bucket = os.environ[\"WORKSPACE_BUCKET\"]\n",
    "ws_bucket_name = re.sub('^gs://', '', ws_bucket)\n",
    "\n",
    "print(f\"workspace name = {ws_name}\")\n",
    "print(f\"workspace project = {ws_project}\")\n",
    "print(f\"workspace bucket = {ws_bucket}\")\n",
    "print(f\"workspace bucket name = {ws_bucket_name}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Inputs:\n",
    "\n",
    "# Mapping specification to evaluate\n",
    "mapping_target = \"anvil\"\n",
    "mapping_target_spec_list = [\"cmg_ext_2\", \"cmg_ext_3\", \"cmg_ext_4\", \"gtex_ext_2\", \"gtex_ext_3\"]\n",
    "\n",
    "# Any known data_file_refs, so file ref fields can be properly evaluated\n",
    "data_file_refs = {   \n",
    "    \"sequencing\": [{\n",
    "        \"column\": \"sequencing_id\",\n",
    "        \"method\": \"file_path_match\",\n",
    "        \"match_multiple_files\": True, \n",
    "        \"match_regex\": None,\n",
    "        \"create_new_field\": True,\n",
    "        \"new_field_name\": \"sequencing_id_file_id\"\n",
    "    }]\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "------------------------------------------------------------------------------------------------------\n",
      "Mapping Compatibility Results for Mapping Target Specifications:\n",
      "------------------------------------------------------------------------------------------------------\n",
      "Target tables not included in specifications:\n",
      "\tcmg_ext_2: anvil_alignmentactivity, anvil_antibody, anvil_assayactivity, anvil_variantcallingactivity\n",
      "\tcmg_ext_3: anvil_alignmentactivity, anvil_antibody, anvil_assayactivity, anvil_variantcallingactivity\n",
      "\tcmg_ext_4: anvil_alignmentactivity, anvil_antibody, anvil_assayactivity, anvil_variantcallingactivity\n",
      "\tgtex_ext_2: anvil_alignmentactivity, anvil_antibody, anvil_assayactivity, anvil_diagnosis, anvil_sequencingactivity, anvil_variantcallingactivity\n",
      "\tgtex_ext_3: anvil_alignmentactivity, anvil_antibody, anvil_assayactivity, anvil_diagnosis, anvil_sequencingactivity, anvil_variantcallingactivity\n",
      "\n",
      "\n",
      "Target fields not included in specification:\n",
      "\tcmg_ext_2: anvil_activity.source_datarepo_row_ids, anvil_activity.used_file_id, anvil_biosample.apriori_cell_type, anvil_biosample.disease, anvil_biosample.source_datarepo_row_ids, anvil_dataset.data_modality, anvil_dataset.principal_investigator, anvil_dataset.source_datarepo_row_ids, anvil_diagnosis.diagnosis_age_lower_bound, anvil_diagnosis.diagnosis_age_unit, anvil_diagnosis.diagnosis_age_upper_bound, anvil_diagnosis.phenopacket, anvil_diagnosis.source_datarepo_row_ids, anvil_donor.genetic_ancestry, anvil_donor.source_datarepo_row_ids, anvil_file.data_modality, anvil_file.is_supplementary, anvil_file.reference_assembly, anvil_file.source_datarepo_row_ids, anvil_project.funded_by, anvil_project.principal_investigator, anvil_project.source_datarepo_row_ids, anvil_sequencingactivity.assay_type, anvil_sequencingactivity.data_modality, anvil_sequencingactivity.source_datarepo_row_ids\n",
      "\tcmg_ext_3: anvil_activity.source_datarepo_row_ids, anvil_activity.used_file_id, anvil_biosample.apriori_cell_type, anvil_biosample.disease, anvil_biosample.source_datarepo_row_ids, anvil_dataset.data_modality, anvil_dataset.principal_investigator, anvil_dataset.source_datarepo_row_ids, anvil_diagnosis.diagnosis_age_lower_bound, anvil_diagnosis.diagnosis_age_unit, anvil_diagnosis.diagnosis_age_upper_bound, anvil_diagnosis.phenopacket, anvil_diagnosis.source_datarepo_row_ids, anvil_donor.genetic_ancestry, anvil_donor.source_datarepo_row_ids, anvil_file.data_modality, anvil_file.is_supplementary, anvil_file.reference_assembly, anvil_file.source_datarepo_row_ids, anvil_project.funded_by, anvil_project.principal_investigator, anvil_project.source_datarepo_row_ids, anvil_sequencingactivity.assay_type, anvil_sequencingactivity.data_modality, anvil_sequencingactivity.source_datarepo_row_ids\n",
      "\tcmg_ext_4: anvil_activity.source_datarepo_row_ids, anvil_activity.used_file_id, anvil_biosample.apriori_cell_type, anvil_biosample.disease, anvil_biosample.source_datarepo_row_ids, anvil_dataset.data_modality, anvil_dataset.principal_investigator, anvil_dataset.source_datarepo_row_ids, anvil_diagnosis.diagnosis_age_lower_bound, anvil_diagnosis.diagnosis_age_unit, anvil_diagnosis.diagnosis_age_upper_bound, anvil_diagnosis.phenopacket, anvil_diagnosis.source_datarepo_row_ids, anvil_donor.genetic_ancestry, anvil_donor.source_datarepo_row_ids, anvil_file.data_modality, anvil_file.is_supplementary, anvil_file.reference_assembly, anvil_file.source_datarepo_row_ids, anvil_project.funded_by, anvil_project.principal_investigator, anvil_project.source_datarepo_row_ids, anvil_sequencingactivity.assay_type, anvil_sequencingactivity.data_modality, anvil_sequencingactivity.source_datarepo_row_ids\n",
      "\tgtex_ext_2: anvil_activity.source_datarepo_row_ids, anvil_activity.used_file_id, anvil_biosample.apriori_cell_type, anvil_biosample.disease, anvil_biosample.source_datarepo_row_ids, anvil_dataset.data_modality, anvil_dataset.principal_investigator, anvil_dataset.source_datarepo_row_ids, anvil_donor.genetic_ancestry, anvil_donor.reported_ethnicity, anvil_donor.source_datarepo_row_ids, anvil_file.data_modality, anvil_file.is_supplementary, anvil_file.reference_assembly, anvil_file.source_datarepo_row_ids, anvil_project.funded_by, anvil_project.principal_investigator, anvil_project.source_datarepo_row_ids\n",
      "\tgtex_ext_3: anvil_activity.source_datarepo_row_ids, anvil_activity.used_file_id, anvil_biosample.apriori_cell_type, anvil_biosample.disease, anvil_biosample.source_datarepo_row_ids, anvil_dataset.data_modality, anvil_dataset.principal_investigator, anvil_dataset.source_datarepo_row_ids, anvil_donor.genetic_ancestry, anvil_donor.reported_ethnicity, anvil_donor.source_datarepo_row_ids, anvil_file.data_modality, anvil_file.is_supplementary, anvil_file.reference_assembly, anvil_file.source_datarepo_row_ids, anvil_project.funded_by, anvil_project.principal_investigator, anvil_project.source_datarepo_row_ids\n",
      "\n",
      "\n",
      "Workspace evaluation against specifications:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: center;\">\n",
       "      <th></th>\n",
       "      <th>workspace</th>\n",
       "      <th>mapping_spec</th>\n",
       "      <th>compatible</th>\n",
       "      <th>can_run_count</th>\n",
       "      <th>can_run_entities</th>\n",
       "      <th>can_fully_run_count</th>\n",
       "      <th>can_fully_run_entities</th>\n",
       "      <th>cnt_valid_attrs</th>\n",
       "      <th>perc_valid_attrs</th>\n",
       "      <th>invalid_attr_list</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>cmg_ext_2</td>\n",
       "      <td>Y</td>\n",
       "      <td>6</td>\n",
       "      <td>{anvil_project, anvil_sequencingactivity, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}</td>\n",
       "      <td>4</td>\n",
       "      <td>{anvil_sequencingactivity, anvil_dataset, anvil_project, anvil_file}</td>\n",
       "      <td>31</td>\n",
       "      <td>0.70</td>\n",
       "      <td>[anvil_donor.phenotypic_sex, anvil_donor.reported_ethnicity, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_diagnosis.onset_age_unit, anvil_diagnosis.phenotype, anvil_diagnosis.onset_age_lower_bound, anvil_diagnosis.onset_age_upper_bound, anvil_diagnosis.disease, anvil_activity.generated_file_id]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>cmg_ext_2</td>\n",
       "      <td>Y</td>\n",
       "      <td>6</td>\n",
       "      <td>{anvil_project, anvil_sequencingactivity, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}</td>\n",
       "      <td>4</td>\n",
       "      <td>{anvil_sequencingactivity, anvil_dataset, anvil_project, anvil_file}</td>\n",
       "      <td>31</td>\n",
       "      <td>0.70</td>\n",
       "      <td>[anvil_donor.phenotypic_sex, anvil_donor.reported_ethnicity, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_diagnosis.onset_age_unit, anvil_diagnosis.phenotype, anvil_diagnosis.onset_age_lower_bound, anvil_diagnosis.onset_age_upper_bound, anvil_diagnosis.disease, anvil_activity.generated_file_id]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>cmg_ext_2</td>\n",
       "      <td>Y</td>\n",
       "      <td>6</td>\n",
       "      <td>{anvil_project, anvil_sequencingactivity, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}</td>\n",
       "      <td>4</td>\n",
       "      <td>{anvil_sequencingactivity, anvil_dataset, anvil_project, anvil_file}</td>\n",
       "      <td>31</td>\n",
       "      <td>0.70</td>\n",
       "      <td>[anvil_donor.phenotypic_sex, anvil_donor.reported_ethnicity, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_diagnosis.onset_age_unit, anvil_diagnosis.phenotype, anvil_diagnosis.onset_age_lower_bound, anvil_diagnosis.onset_age_upper_bound, anvil_diagnosis.disease, anvil_activity.generated_file_id]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>cmg_ext_2</td>\n",
       "      <td>Y</td>\n",
       "      <td>6</td>\n",
       "      <td>{anvil_project, anvil_sequencingactivity, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}</td>\n",
       "      <td>4</td>\n",
       "      <td>{anvil_sequencingactivity, anvil_dataset, anvil_project, anvil_file}</td>\n",
       "      <td>31</td>\n",
       "      <td>0.70</td>\n",
       "      <td>[anvil_donor.phenotypic_sex, anvil_donor.reported_ethnicity, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_diagnosis.onset_age_unit, anvil_diagnosis.phenotype, anvil_diagnosis.onset_age_lower_bound, anvil_diagnosis.onset_age_upper_bound, anvil_diagnosis.disease, anvil_activity.generated_file_id]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>cmg_ext_2</td>\n",
       "      <td>Y</td>\n",
       "      <td>6</td>\n",
       "      <td>{anvil_project, anvil_sequencingactivity, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}</td>\n",
       "      <td>4</td>\n",
       "      <td>{anvil_sequencingactivity, anvil_dataset, anvil_project, anvil_file}</td>\n",
       "      <td>31</td>\n",
       "      <td>0.70</td>\n",
       "      <td>[anvil_donor.phenotypic_sex, anvil_donor.reported_ethnicity, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_diagnosis.onset_age_unit, anvil_diagnosis.phenotype, anvil_diagnosis.onset_age_lower_bound, anvil_diagnosis.onset_age_upper_bound, anvil_diagnosis.disease, anvil_activity.generated_file_id]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>cmg_ext_2</td>\n",
       "      <td>Y</td>\n",
       "      <td>6</td>\n",
       "      <td>{anvil_project, anvil_sequencingactivity, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}</td>\n",
       "      <td>4</td>\n",
       "      <td>{anvil_sequencingactivity, anvil_dataset, anvil_project, anvil_file}</td>\n",
       "      <td>31</td>\n",
       "      <td>0.70</td>\n",
       "      <td>[anvil_donor.phenotypic_sex, anvil_donor.reported_ethnicity, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_diagnosis.onset_age_unit, anvil_diagnosis.phenotype, anvil_diagnosis.onset_age_lower_bound, anvil_diagnosis.onset_age_upper_bound, anvil_diagnosis.disease, anvil_activity.generated_file_id]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>cmg_ext_2</td>\n",
       "      <td>Y</td>\n",
       "      <td>6</td>\n",
       "      <td>{anvil_project, anvil_sequencingactivity, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}</td>\n",
       "      <td>4</td>\n",
       "      <td>{anvil_sequencingactivity, anvil_dataset, anvil_project, anvil_file}</td>\n",
       "      <td>31</td>\n",
       "      <td>0.70</td>\n",
       "      <td>[anvil_donor.phenotypic_sex, anvil_donor.reported_ethnicity, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_diagnosis.onset_age_unit, anvil_diagnosis.phenotype, anvil_diagnosis.onset_age_lower_bound, anvil_diagnosis.onset_age_upper_bound, anvil_diagnosis.disease, anvil_activity.generated_file_id]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>cmg_ext_3</td>\n",
       "      <td>Y</td>\n",
       "      <td>6</td>\n",
       "      <td>{anvil_project, anvil_sequencingactivity, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}</td>\n",
       "      <td>4</td>\n",
       "      <td>{anvil_sequencingactivity, anvil_dataset, anvil_project, anvil_file}</td>\n",
       "      <td>31</td>\n",
       "      <td>0.70</td>\n",
       "      <td>[anvil_donor.phenotypic_sex, anvil_donor.reported_ethnicity, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_diagnosis.onset_age_unit, anvil_diagnosis.phenotype, anvil_diagnosis.onset_age_lower_bound, anvil_diagnosis.onset_age_upper_bound, anvil_diagnosis.disease, anvil_activity.generated_file_id]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>cmg_ext_3</td>\n",
       "      <td>Y</td>\n",
       "      <td>6</td>\n",
       "      <td>{anvil_project, anvil_sequencingactivity, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}</td>\n",
       "      <td>4</td>\n",
       "      <td>{anvil_sequencingactivity, anvil_dataset, anvil_project, anvil_file}</td>\n",
       "      <td>31</td>\n",
       "      <td>0.70</td>\n",
       "      <td>[anvil_donor.phenotypic_sex, anvil_donor.reported_ethnicity, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_diagnosis.onset_age_unit, anvil_diagnosis.phenotype, anvil_diagnosis.onset_age_lower_bound, anvil_diagnosis.onset_age_upper_bound, anvil_diagnosis.disease, anvil_activity.generated_file_id]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>cmg_ext_3</td>\n",
       "      <td>Y</td>\n",
       "      <td>6</td>\n",
       "      <td>{anvil_project, anvil_sequencingactivity, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}</td>\n",
       "      <td>4</td>\n",
       "      <td>{anvil_sequencingactivity, anvil_dataset, anvil_project, anvil_file}</td>\n",
       "      <td>31</td>\n",
       "      <td>0.70</td>\n",
       "      <td>[anvil_donor.phenotypic_sex, anvil_donor.reported_ethnicity, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_diagnosis.onset_age_unit, anvil_diagnosis.phenotype, anvil_diagnosis.onset_age_lower_bound, anvil_diagnosis.onset_age_upper_bound, anvil_diagnosis.disease, anvil_activity.generated_file_id]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>cmg_ext_3</td>\n",
       "      <td>Y</td>\n",
       "      <td>6</td>\n",
       "      <td>{anvil_project, anvil_sequencingactivity, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}</td>\n",
       "      <td>4</td>\n",
       "      <td>{anvil_sequencingactivity, anvil_dataset, anvil_project, anvil_file}</td>\n",
       "      <td>31</td>\n",
       "      <td>0.70</td>\n",
       "      <td>[anvil_donor.phenotypic_sex, anvil_donor.reported_ethnicity, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_diagnosis.onset_age_unit, anvil_diagnosis.phenotype, anvil_diagnosis.onset_age_lower_bound, anvil_diagnosis.onset_age_upper_bound, anvil_diagnosis.disease, anvil_activity.generated_file_id]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>cmg_ext_3</td>\n",
       "      <td>Y</td>\n",
       "      <td>6</td>\n",
       "      <td>{anvil_project, anvil_sequencingactivity, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}</td>\n",
       "      <td>4</td>\n",
       "      <td>{anvil_sequencingactivity, anvil_dataset, anvil_project, anvil_file}</td>\n",
       "      <td>31</td>\n",
       "      <td>0.70</td>\n",
       "      <td>[anvil_donor.phenotypic_sex, anvil_donor.reported_ethnicity, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_diagnosis.onset_age_unit, anvil_diagnosis.phenotype, anvil_diagnosis.onset_age_lower_bound, anvil_diagnosis.onset_age_upper_bound, anvil_diagnosis.disease, anvil_activity.generated_file_id]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>cmg_ext_3</td>\n",
       "      <td>Y</td>\n",
       "      <td>6</td>\n",
       "      <td>{anvil_project, anvil_sequencingactivity, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}</td>\n",
       "      <td>4</td>\n",
       "      <td>{anvil_sequencingactivity, anvil_dataset, anvil_project, anvil_file}</td>\n",
       "      <td>31</td>\n",
       "      <td>0.70</td>\n",
       "      <td>[anvil_donor.phenotypic_sex, anvil_donor.reported_ethnicity, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_diagnosis.onset_age_unit, anvil_diagnosis.phenotype, anvil_diagnosis.onset_age_lower_bound, anvil_diagnosis.onset_age_upper_bound, anvil_diagnosis.disease, anvil_activity.generated_file_id]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>cmg_ext_3</td>\n",
       "      <td>Y</td>\n",
       "      <td>6</td>\n",
       "      <td>{anvil_project, anvil_sequencingactivity, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}</td>\n",
       "      <td>4</td>\n",
       "      <td>{anvil_sequencingactivity, anvil_dataset, anvil_project, anvil_file}</td>\n",
       "      <td>31</td>\n",
       "      <td>0.70</td>\n",
       "      <td>[anvil_donor.phenotypic_sex, anvil_donor.reported_ethnicity, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_diagnosis.onset_age_unit, anvil_diagnosis.phenotype, anvil_diagnosis.onset_age_lower_bound, anvil_diagnosis.onset_age_upper_bound, anvil_diagnosis.disease, anvil_activity.generated_file_id]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>cmg_ext_4</td>\n",
       "      <td>Y</td>\n",
       "      <td>6</td>\n",
       "      <td>{anvil_project, anvil_sequencingactivity, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}</td>\n",
       "      <td>4</td>\n",
       "      <td>{anvil_sequencingactivity, anvil_dataset, anvil_project, anvil_file}</td>\n",
       "      <td>31</td>\n",
       "      <td>0.70</td>\n",
       "      <td>[anvil_donor.phenotypic_sex, anvil_donor.reported_ethnicity, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_diagnosis.onset_age_unit, anvil_diagnosis.phenotype, anvil_diagnosis.onset_age_lower_bound, anvil_diagnosis.onset_age_upper_bound, anvil_diagnosis.disease, anvil_activity.generated_file_id]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>cmg_ext_4</td>\n",
       "      <td>Y</td>\n",
       "      <td>6</td>\n",
       "      <td>{anvil_project, anvil_sequencingactivity, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}</td>\n",
       "      <td>4</td>\n",
       "      <td>{anvil_sequencingactivity, anvil_dataset, anvil_project, anvil_file}</td>\n",
       "      <td>31</td>\n",
       "      <td>0.70</td>\n",
       "      <td>[anvil_donor.phenotypic_sex, anvil_donor.reported_ethnicity, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_diagnosis.onset_age_unit, anvil_diagnosis.phenotype, anvil_diagnosis.onset_age_lower_bound, anvil_diagnosis.onset_age_upper_bound, anvil_diagnosis.disease, anvil_activity.generated_file_id]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>cmg_ext_4</td>\n",
       "      <td>Y</td>\n",
       "      <td>6</td>\n",
       "      <td>{anvil_project, anvil_sequencingactivity, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}</td>\n",
       "      <td>4</td>\n",
       "      <td>{anvil_sequencingactivity, anvil_dataset, anvil_project, anvil_file}</td>\n",
       "      <td>31</td>\n",
       "      <td>0.70</td>\n",
       "      <td>[anvil_donor.phenotypic_sex, anvil_donor.reported_ethnicity, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_diagnosis.onset_age_unit, anvil_diagnosis.phenotype, anvil_diagnosis.onset_age_lower_bound, anvil_diagnosis.onset_age_upper_bound, anvil_diagnosis.disease, anvil_activity.generated_file_id]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>cmg_ext_4</td>\n",
       "      <td>Y</td>\n",
       "      <td>6</td>\n",
       "      <td>{anvil_project, anvil_sequencingactivity, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}</td>\n",
       "      <td>4</td>\n",
       "      <td>{anvil_sequencingactivity, anvil_dataset, anvil_project, anvil_file}</td>\n",
       "      <td>31</td>\n",
       "      <td>0.70</td>\n",
       "      <td>[anvil_donor.phenotypic_sex, anvil_donor.reported_ethnicity, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_diagnosis.onset_age_unit, anvil_diagnosis.phenotype, anvil_diagnosis.onset_age_lower_bound, anvil_diagnosis.onset_age_upper_bound, anvil_diagnosis.disease, anvil_activity.generated_file_id]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>cmg_ext_4</td>\n",
       "      <td>Y</td>\n",
       "      <td>6</td>\n",
       "      <td>{anvil_project, anvil_sequencingactivity, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}</td>\n",
       "      <td>4</td>\n",
       "      <td>{anvil_sequencingactivity, anvil_dataset, anvil_project, anvil_file}</td>\n",
       "      <td>31</td>\n",
       "      <td>0.70</td>\n",
       "      <td>[anvil_donor.phenotypic_sex, anvil_donor.reported_ethnicity, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_diagnosis.onset_age_unit, anvil_diagnosis.phenotype, anvil_diagnosis.onset_age_lower_bound, anvil_diagnosis.onset_age_upper_bound, anvil_diagnosis.disease, anvil_activity.generated_file_id]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>cmg_ext_4</td>\n",
       "      <td>Y</td>\n",
       "      <td>6</td>\n",
       "      <td>{anvil_project, anvil_sequencingactivity, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}</td>\n",
       "      <td>4</td>\n",
       "      <td>{anvil_sequencingactivity, anvil_dataset, anvil_project, anvil_file}</td>\n",
       "      <td>31</td>\n",
       "      <td>0.70</td>\n",
       "      <td>[anvil_donor.phenotypic_sex, anvil_donor.reported_ethnicity, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_diagnosis.onset_age_unit, anvil_diagnosis.phenotype, anvil_diagnosis.onset_age_lower_bound, anvil_diagnosis.onset_age_upper_bound, anvil_diagnosis.disease, anvil_activity.generated_file_id]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>cmg_ext_4</td>\n",
       "      <td>Y</td>\n",
       "      <td>6</td>\n",
       "      <td>{anvil_project, anvil_sequencingactivity, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}</td>\n",
       "      <td>4</td>\n",
       "      <td>{anvil_sequencingactivity, anvil_dataset, anvil_project, anvil_file}</td>\n",
       "      <td>31</td>\n",
       "      <td>0.70</td>\n",
       "      <td>[anvil_donor.phenotypic_sex, anvil_donor.reported_ethnicity, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_diagnosis.onset_age_unit, anvil_diagnosis.phenotype, anvil_diagnosis.onset_age_lower_bound, anvil_diagnosis.onset_age_upper_bound, anvil_diagnosis.disease, anvil_activity.generated_file_id]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>gtex_ext_2</td>\n",
       "      <td>N</td>\n",
       "      <td>5</td>\n",
       "      <td>{anvil_project, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}</td>\n",
       "      <td>3</td>\n",
       "      <td>{anvil_dataset, anvil_project, anvil_file}</td>\n",
       "      <td>25</td>\n",
       "      <td>0.78</td>\n",
       "      <td>[anvil_donor.phenotypic_sex, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_id, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_activity.generated_file_id]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>gtex_ext_2</td>\n",
       "      <td>N</td>\n",
       "      <td>5</td>\n",
       "      <td>{anvil_project, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}</td>\n",
       "      <td>3</td>\n",
       "      <td>{anvil_dataset, anvil_project, anvil_file}</td>\n",
       "      <td>25</td>\n",
       "      <td>0.78</td>\n",
       "      <td>[anvil_donor.phenotypic_sex, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_id, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_activity.generated_file_id]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>gtex_ext_2</td>\n",
       "      <td>N</td>\n",
       "      <td>4</td>\n",
       "      <td>{anvil_biosample, anvil_dataset, anvil_project, anvil_file}</td>\n",
       "      <td>3</td>\n",
       "      <td>{anvil_dataset, anvil_project, anvil_file}</td>\n",
       "      <td>23</td>\n",
       "      <td>0.72</td>\n",
       "      <td>[anvil_donor.phenotypic_sex, anvil_donor.donor_id, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_id, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.biosample_id, anvil_biosample.donor_age_at_collection_lower_bound, anvil_activity.used_biosample_id, anvil_activity.activity_id, anvil_activity.generated_file_id]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>gtex_ext_2</td>\n",
       "      <td>N</td>\n",
       "      <td>4</td>\n",
       "      <td>{anvil_biosample, anvil_dataset, anvil_project, anvil_file}</td>\n",
       "      <td>3</td>\n",
       "      <td>{anvil_dataset, anvil_project, anvil_file}</td>\n",
       "      <td>23</td>\n",
       "      <td>0.72</td>\n",
       "      <td>[anvil_donor.phenotypic_sex, anvil_donor.donor_id, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_id, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.biosample_id, anvil_biosample.donor_age_at_collection_lower_bound, anvil_activity.used_biosample_id, anvil_activity.activity_id, anvil_activity.generated_file_id]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>gtex_ext_2</td>\n",
       "      <td>N</td>\n",
       "      <td>4</td>\n",
       "      <td>{anvil_biosample, anvil_dataset, anvil_project, anvil_file}</td>\n",
       "      <td>3</td>\n",
       "      <td>{anvil_dataset, anvil_project, anvil_file}</td>\n",
       "      <td>23</td>\n",
       "      <td>0.72</td>\n",
       "      <td>[anvil_donor.phenotypic_sex, anvil_donor.donor_id, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_id, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.biosample_id, anvil_biosample.donor_age_at_collection_lower_bound, anvil_activity.used_biosample_id, anvil_activity.activity_id, anvil_activity.generated_file_id]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>gtex_ext_2</td>\n",
       "      <td>N</td>\n",
       "      <td>4</td>\n",
       "      <td>{anvil_biosample, anvil_dataset, anvil_project, anvil_file}</td>\n",
       "      <td>3</td>\n",
       "      <td>{anvil_dataset, anvil_project, anvil_file}</td>\n",
       "      <td>23</td>\n",
       "      <td>0.72</td>\n",
       "      <td>[anvil_donor.phenotypic_sex, anvil_donor.donor_id, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_id, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.biosample_id, anvil_biosample.donor_age_at_collection_lower_bound, anvil_activity.used_biosample_id, anvil_activity.activity_id, anvil_activity.generated_file_id]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>gtex_ext_2</td>\n",
       "      <td>N</td>\n",
       "      <td>4</td>\n",
       "      <td>{anvil_biosample, anvil_dataset, anvil_project, anvil_file}</td>\n",
       "      <td>3</td>\n",
       "      <td>{anvil_dataset, anvil_project, anvil_file}</td>\n",
       "      <td>23</td>\n",
       "      <td>0.72</td>\n",
       "      <td>[anvil_donor.phenotypic_sex, anvil_donor.donor_id, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_id, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.biosample_id, anvil_biosample.donor_age_at_collection_lower_bound, anvil_activity.used_biosample_id, anvil_activity.activity_id, anvil_activity.generated_file_id]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>gtex_ext_3</td>\n",
       "      <td>N</td>\n",
       "      <td>5</td>\n",
       "      <td>{anvil_project, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}</td>\n",
       "      <td>3</td>\n",
       "      <td>{anvil_dataset, anvil_project, anvil_file}</td>\n",
       "      <td>25</td>\n",
       "      <td>0.78</td>\n",
       "      <td>[anvil_donor.phenotypic_sex, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_activity.generated_file_id]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>gtex_ext_3</td>\n",
       "      <td>N</td>\n",
       "      <td>5</td>\n",
       "      <td>{anvil_project, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}</td>\n",
       "      <td>3</td>\n",
       "      <td>{anvil_dataset, anvil_project, anvil_file}</td>\n",
       "      <td>25</td>\n",
       "      <td>0.78</td>\n",
       "      <td>[anvil_donor.phenotypic_sex, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_activity.generated_file_id]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>gtex_ext_3</td>\n",
       "      <td>N</td>\n",
       "      <td>4</td>\n",
       "      <td>{anvil_biosample, anvil_dataset, anvil_project, anvil_file}</td>\n",
       "      <td>3</td>\n",
       "      <td>{anvil_dataset, anvil_project, anvil_file}</td>\n",
       "      <td>24</td>\n",
       "      <td>0.75</td>\n",
       "      <td>[anvil_donor.phenotypic_sex, anvil_donor.donor_id, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_id, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.biosample_id, anvil_biosample.donor_age_at_collection_lower_bound, anvil_activity.used_biosample_id, anvil_activity.activity_id, anvil_activity.generated_file_id]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>gtex_ext_3</td>\n",
       "      <td>N</td>\n",
       "      <td>4</td>\n",
       "      <td>{anvil_biosample, anvil_dataset, anvil_project, anvil_file}</td>\n",
       "      <td>3</td>\n",
       "      <td>{anvil_dataset, anvil_project, anvil_file}</td>\n",
       "      <td>24</td>\n",
       "      <td>0.75</td>\n",
       "      <td>[anvil_donor.phenotypic_sex, anvil_donor.donor_id, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_id, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.biosample_id, anvil_biosample.donor_age_at_collection_lower_bound, anvil_activity.used_biosample_id, anvil_activity.activity_id, anvil_activity.generated_file_id]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>gtex_ext_3</td>\n",
       "      <td>N</td>\n",
       "      <td>4</td>\n",
       "      <td>{anvil_biosample, anvil_dataset, anvil_project, anvil_file}</td>\n",
       "      <td>3</td>\n",
       "      <td>{anvil_dataset, anvil_project, anvil_file}</td>\n",
       "      <td>24</td>\n",
       "      <td>0.75</td>\n",
       "      <td>[anvil_donor.phenotypic_sex, anvil_donor.donor_id, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_id, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.biosample_id, anvil_biosample.donor_age_at_collection_lower_bound, anvil_activity.used_biosample_id, anvil_activity.activity_id, anvil_activity.generated_file_id]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>gtex_ext_3</td>\n",
       "      <td>N</td>\n",
       "      <td>4</td>\n",
       "      <td>{anvil_biosample, anvil_dataset, anvil_project, anvil_file}</td>\n",
       "      <td>3</td>\n",
       "      <td>{anvil_dataset, anvil_project, anvil_file}</td>\n",
       "      <td>24</td>\n",
       "      <td>0.75</td>\n",
       "      <td>[anvil_donor.phenotypic_sex, anvil_donor.donor_id, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_id, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.biosample_id, anvil_biosample.donor_age_at_collection_lower_bound, anvil_activity.used_biosample_id, anvil_activity.activity_id, anvil_activity.generated_file_id]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>gtex_ext_3</td>\n",
       "      <td>N</td>\n",
       "      <td>4</td>\n",
       "      <td>{anvil_biosample, anvil_dataset, anvil_project, anvil_file}</td>\n",
       "      <td>3</td>\n",
       "      <td>{anvil_dataset, anvil_project, anvil_file}</td>\n",
       "      <td>24</td>\n",
       "      <td>0.75</td>\n",
       "      <td>[anvil_donor.phenotypic_sex, anvil_donor.donor_id, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_id, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.biosample_id, anvil_biosample.donor_age_at_collection_lower_bound, anvil_activity.used_biosample_id, anvil_activity.activity_id, anvil_activity.generated_file_id]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     workspace                   mapping_spec compatible  can_run_count                                           can_run_entities                                           can_fully_run_count                         can_fully_run_entities                         cnt_valid_attrs  perc_valid_attrs                                                                                                                                                                                                                                     invalid_attr_list                                                                                                                                                                                                                                     \n",
       "0                anvil_cmg_broad_brain_engle_wes   cmg_ext_2       Y            6        {anvil_project, anvil_sequencingactivity, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}           4           {anvil_sequencingactivity, anvil_dataset, anvil_project, anvil_file}        31               0.70        [anvil_donor.phenotypic_sex, anvil_donor.reported_ethnicity, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_diagnosis.onset_age_unit, anvil_diagnosis.phenotype, anvil_diagnosis.onset_age_lower_bound, anvil_diagnosis.onset_age_upper_bound, anvil_diagnosis.disease, anvil_activity.generated_file_id]\n",
       "1     anvil_cmg_broad_genitourinary_sinclair_wes   cmg_ext_2       Y            6        {anvil_project, anvil_sequencingactivity, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}           4           {anvil_sequencingactivity, anvil_dataset, anvil_project, anvil_file}        31               0.70        [anvil_donor.phenotypic_sex, anvil_donor.reported_ethnicity, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_diagnosis.onset_age_unit, anvil_diagnosis.phenotype, anvil_diagnosis.onset_age_lower_bound, anvil_diagnosis.onset_age_upper_bound, anvil_diagnosis.disease, anvil_activity.generated_file_id]\n",
       "2            anvil_cmg_broad_orphan_jueppner_wes   cmg_ext_2       Y            6        {anvil_project, anvil_sequencingactivity, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}           4           {anvil_sequencingactivity, anvil_dataset, anvil_project, anvil_file}        31               0.70        [anvil_donor.phenotypic_sex, anvil_donor.reported_ethnicity, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_diagnosis.onset_age_unit, anvil_diagnosis.phenotype, anvil_diagnosis.onset_age_lower_bound, anvil_diagnosis.onset_age_upper_bound, anvil_diagnosis.disease, anvil_activity.generated_file_id]\n",
       "3   anvil_cmg_broad_genitourinary_hirschhorn_wes   cmg_ext_2       Y            6        {anvil_project, anvil_sequencingactivity, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}           4           {anvil_sequencingactivity, anvil_dataset, anvil_project, anvil_file}        31               0.70        [anvil_donor.phenotypic_sex, anvil_donor.reported_ethnicity, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_diagnosis.onset_age_unit, anvil_diagnosis.phenotype, anvil_diagnosis.onset_age_lower_bound, anvil_diagnosis.onset_age_upper_bound, anvil_diagnosis.disease, anvil_activity.generated_file_id]\n",
       "4               anvil_cmg_broad_orphan_scott_wes   cmg_ext_2       Y            6        {anvil_project, anvil_sequencingactivity, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}           4           {anvil_sequencingactivity, anvil_dataset, anvil_project, anvil_file}        31               0.70        [anvil_donor.phenotypic_sex, anvil_donor.reported_ethnicity, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_diagnosis.onset_age_unit, anvil_diagnosis.phenotype, anvil_diagnosis.onset_age_lower_bound, anvil_diagnosis.onset_age_upper_bound, anvil_diagnosis.disease, anvil_activity.generated_file_id]\n",
       "5          anvil_cmg_broad_orphan_vcgs-white_wes   cmg_ext_2       Y            6        {anvil_project, anvil_sequencingactivity, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}           4           {anvil_sequencingactivity, anvil_dataset, anvil_project, anvil_file}        31               0.70        [anvil_donor.phenotypic_sex, anvil_donor.reported_ethnicity, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_diagnosis.onset_age_unit, anvil_diagnosis.phenotype, anvil_diagnosis.onset_age_lower_bound, anvil_diagnosis.onset_age_upper_bound, anvil_diagnosis.disease, anvil_activity.generated_file_id]\n",
       "6    anvil_cmg_broad_stillbirth_wilkins-haug_wes   cmg_ext_2       Y            6        {anvil_project, anvil_sequencingactivity, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}           4           {anvil_sequencingactivity, anvil_dataset, anvil_project, anvil_file}        31               0.70        [anvil_donor.phenotypic_sex, anvil_donor.reported_ethnicity, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_diagnosis.onset_age_unit, anvil_diagnosis.phenotype, anvil_diagnosis.onset_age_lower_bound, anvil_diagnosis.onset_age_upper_bound, anvil_diagnosis.disease, anvil_activity.generated_file_id]\n",
       "7                anvil_cmg_broad_brain_engle_wes   cmg_ext_3       Y            6        {anvil_project, anvil_sequencingactivity, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}           4           {anvil_sequencingactivity, anvil_dataset, anvil_project, anvil_file}        31               0.70        [anvil_donor.phenotypic_sex, anvil_donor.reported_ethnicity, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_diagnosis.onset_age_unit, anvil_diagnosis.phenotype, anvil_diagnosis.onset_age_lower_bound, anvil_diagnosis.onset_age_upper_bound, anvil_diagnosis.disease, anvil_activity.generated_file_id]\n",
       "8     anvil_cmg_broad_genitourinary_sinclair_wes   cmg_ext_3       Y            6        {anvil_project, anvil_sequencingactivity, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}           4           {anvil_sequencingactivity, anvil_dataset, anvil_project, anvil_file}        31               0.70        [anvil_donor.phenotypic_sex, anvil_donor.reported_ethnicity, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_diagnosis.onset_age_unit, anvil_diagnosis.phenotype, anvil_diagnosis.onset_age_lower_bound, anvil_diagnosis.onset_age_upper_bound, anvil_diagnosis.disease, anvil_activity.generated_file_id]\n",
       "9            anvil_cmg_broad_orphan_jueppner_wes   cmg_ext_3       Y            6        {anvil_project, anvil_sequencingactivity, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}           4           {anvil_sequencingactivity, anvil_dataset, anvil_project, anvil_file}        31               0.70        [anvil_donor.phenotypic_sex, anvil_donor.reported_ethnicity, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_diagnosis.onset_age_unit, anvil_diagnosis.phenotype, anvil_diagnosis.onset_age_lower_bound, anvil_diagnosis.onset_age_upper_bound, anvil_diagnosis.disease, anvil_activity.generated_file_id]\n",
       "10  anvil_cmg_broad_genitourinary_hirschhorn_wes   cmg_ext_3       Y            6        {anvil_project, anvil_sequencingactivity, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}           4           {anvil_sequencingactivity, anvil_dataset, anvil_project, anvil_file}        31               0.70        [anvil_donor.phenotypic_sex, anvil_donor.reported_ethnicity, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_diagnosis.onset_age_unit, anvil_diagnosis.phenotype, anvil_diagnosis.onset_age_lower_bound, anvil_diagnosis.onset_age_upper_bound, anvil_diagnosis.disease, anvil_activity.generated_file_id]\n",
       "11              anvil_cmg_broad_orphan_scott_wes   cmg_ext_3       Y            6        {anvil_project, anvil_sequencingactivity, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}           4           {anvil_sequencingactivity, anvil_dataset, anvil_project, anvil_file}        31               0.70        [anvil_donor.phenotypic_sex, anvil_donor.reported_ethnicity, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_diagnosis.onset_age_unit, anvil_diagnosis.phenotype, anvil_diagnosis.onset_age_lower_bound, anvil_diagnosis.onset_age_upper_bound, anvil_diagnosis.disease, anvil_activity.generated_file_id]\n",
       "12         anvil_cmg_broad_orphan_vcgs-white_wes   cmg_ext_3       Y            6        {anvil_project, anvil_sequencingactivity, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}           4           {anvil_sequencingactivity, anvil_dataset, anvil_project, anvil_file}        31               0.70        [anvil_donor.phenotypic_sex, anvil_donor.reported_ethnicity, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_diagnosis.onset_age_unit, anvil_diagnosis.phenotype, anvil_diagnosis.onset_age_lower_bound, anvil_diagnosis.onset_age_upper_bound, anvil_diagnosis.disease, anvil_activity.generated_file_id]\n",
       "13   anvil_cmg_broad_stillbirth_wilkins-haug_wes   cmg_ext_3       Y            6        {anvil_project, anvil_sequencingactivity, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}           4           {anvil_sequencingactivity, anvil_dataset, anvil_project, anvil_file}        31               0.70        [anvil_donor.phenotypic_sex, anvil_donor.reported_ethnicity, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_diagnosis.onset_age_unit, anvil_diagnosis.phenotype, anvil_diagnosis.onset_age_lower_bound, anvil_diagnosis.onset_age_upper_bound, anvil_diagnosis.disease, anvil_activity.generated_file_id]\n",
       "14               anvil_cmg_broad_brain_engle_wes   cmg_ext_4       Y            6        {anvil_project, anvil_sequencingactivity, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}           4           {anvil_sequencingactivity, anvil_dataset, anvil_project, anvil_file}        31               0.70        [anvil_donor.phenotypic_sex, anvil_donor.reported_ethnicity, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_diagnosis.onset_age_unit, anvil_diagnosis.phenotype, anvil_diagnosis.onset_age_lower_bound, anvil_diagnosis.onset_age_upper_bound, anvil_diagnosis.disease, anvil_activity.generated_file_id]\n",
       "15    anvil_cmg_broad_genitourinary_sinclair_wes   cmg_ext_4       Y            6        {anvil_project, anvil_sequencingactivity, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}           4           {anvil_sequencingactivity, anvil_dataset, anvil_project, anvil_file}        31               0.70        [anvil_donor.phenotypic_sex, anvil_donor.reported_ethnicity, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_diagnosis.onset_age_unit, anvil_diagnosis.phenotype, anvil_diagnosis.onset_age_lower_bound, anvil_diagnosis.onset_age_upper_bound, anvil_diagnosis.disease, anvil_activity.generated_file_id]\n",
       "16           anvil_cmg_broad_orphan_jueppner_wes   cmg_ext_4       Y            6        {anvil_project, anvil_sequencingactivity, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}           4           {anvil_sequencingactivity, anvil_dataset, anvil_project, anvil_file}        31               0.70        [anvil_donor.phenotypic_sex, anvil_donor.reported_ethnicity, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_diagnosis.onset_age_unit, anvil_diagnosis.phenotype, anvil_diagnosis.onset_age_lower_bound, anvil_diagnosis.onset_age_upper_bound, anvil_diagnosis.disease, anvil_activity.generated_file_id]\n",
       "17  anvil_cmg_broad_genitourinary_hirschhorn_wes   cmg_ext_4       Y            6        {anvil_project, anvil_sequencingactivity, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}           4           {anvil_sequencingactivity, anvil_dataset, anvil_project, anvil_file}        31               0.70        [anvil_donor.phenotypic_sex, anvil_donor.reported_ethnicity, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_diagnosis.onset_age_unit, anvil_diagnosis.phenotype, anvil_diagnosis.onset_age_lower_bound, anvil_diagnosis.onset_age_upper_bound, anvil_diagnosis.disease, anvil_activity.generated_file_id]\n",
       "18              anvil_cmg_broad_orphan_scott_wes   cmg_ext_4       Y            6        {anvil_project, anvil_sequencingactivity, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}           4           {anvil_sequencingactivity, anvil_dataset, anvil_project, anvil_file}        31               0.70        [anvil_donor.phenotypic_sex, anvil_donor.reported_ethnicity, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_diagnosis.onset_age_unit, anvil_diagnosis.phenotype, anvil_diagnosis.onset_age_lower_bound, anvil_diagnosis.onset_age_upper_bound, anvil_diagnosis.disease, anvil_activity.generated_file_id]\n",
       "19         anvil_cmg_broad_orphan_vcgs-white_wes   cmg_ext_4       Y            6        {anvil_project, anvil_sequencingactivity, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}           4           {anvil_sequencingactivity, anvil_dataset, anvil_project, anvil_file}        31               0.70        [anvil_donor.phenotypic_sex, anvil_donor.reported_ethnicity, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_diagnosis.onset_age_unit, anvil_diagnosis.phenotype, anvil_diagnosis.onset_age_lower_bound, anvil_diagnosis.onset_age_upper_bound, anvil_diagnosis.disease, anvil_activity.generated_file_id]\n",
       "20   anvil_cmg_broad_stillbirth_wilkins-haug_wes   cmg_ext_4       Y            6        {anvil_project, anvil_sequencingactivity, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}           4           {anvil_sequencingactivity, anvil_dataset, anvil_project, anvil_file}        31               0.70        [anvil_donor.phenotypic_sex, anvil_donor.reported_ethnicity, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_diagnosis.onset_age_unit, anvil_diagnosis.phenotype, anvil_diagnosis.onset_age_lower_bound, anvil_diagnosis.onset_age_upper_bound, anvil_diagnosis.disease, anvil_activity.generated_file_id]\n",
       "21  anvil_cmg_broad_genitourinary_hirschhorn_wes  gtex_ext_2       N            5                                  {anvil_project, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}           3                                     {anvil_dataset, anvil_project, anvil_file}        25               0.78                                                                                                                                                                                [anvil_donor.phenotypic_sex, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_id, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_activity.generated_file_id]\n",
       "22         anvil_cmg_broad_orphan_vcgs-white_wes  gtex_ext_2       N            5                                  {anvil_project, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}           3                                     {anvil_dataset, anvil_project, anvil_file}        25               0.78                                                                                                                                                                                [anvil_donor.phenotypic_sex, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_id, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_activity.generated_file_id]\n",
       "23               anvil_cmg_broad_brain_engle_wes  gtex_ext_2       N            4                                               {anvil_biosample, anvil_dataset, anvil_project, anvil_file}           3                                     {anvil_dataset, anvil_project, anvil_file}        23               0.72                                                              [anvil_donor.phenotypic_sex, anvil_donor.donor_id, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_id, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.biosample_id, anvil_biosample.donor_age_at_collection_lower_bound, anvil_activity.used_biosample_id, anvil_activity.activity_id, anvil_activity.generated_file_id]\n",
       "24    anvil_cmg_broad_genitourinary_sinclair_wes  gtex_ext_2       N            4                                               {anvil_biosample, anvil_dataset, anvil_project, anvil_file}           3                                     {anvil_dataset, anvil_project, anvil_file}        23               0.72                                                              [anvil_donor.phenotypic_sex, anvil_donor.donor_id, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_id, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.biosample_id, anvil_biosample.donor_age_at_collection_lower_bound, anvil_activity.used_biosample_id, anvil_activity.activity_id, anvil_activity.generated_file_id]\n",
       "25           anvil_cmg_broad_orphan_jueppner_wes  gtex_ext_2       N            4                                               {anvil_biosample, anvil_dataset, anvil_project, anvil_file}           3                                     {anvil_dataset, anvil_project, anvil_file}        23               0.72                                                              [anvil_donor.phenotypic_sex, anvil_donor.donor_id, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_id, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.biosample_id, anvil_biosample.donor_age_at_collection_lower_bound, anvil_activity.used_biosample_id, anvil_activity.activity_id, anvil_activity.generated_file_id]\n",
       "26              anvil_cmg_broad_orphan_scott_wes  gtex_ext_2       N            4                                               {anvil_biosample, anvil_dataset, anvil_project, anvil_file}           3                                     {anvil_dataset, anvil_project, anvil_file}        23               0.72                                                              [anvil_donor.phenotypic_sex, anvil_donor.donor_id, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_id, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.biosample_id, anvil_biosample.donor_age_at_collection_lower_bound, anvil_activity.used_biosample_id, anvil_activity.activity_id, anvil_activity.generated_file_id]\n",
       "27   anvil_cmg_broad_stillbirth_wilkins-haug_wes  gtex_ext_2       N            4                                               {anvil_biosample, anvil_dataset, anvil_project, anvil_file}           3                                     {anvil_dataset, anvil_project, anvil_file}        23               0.72                                                              [anvil_donor.phenotypic_sex, anvil_donor.donor_id, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_id, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.biosample_id, anvil_biosample.donor_age_at_collection_lower_bound, anvil_activity.used_biosample_id, anvil_activity.activity_id, anvil_activity.generated_file_id]\n",
       "28  anvil_cmg_broad_genitourinary_hirschhorn_wes  gtex_ext_3       N            5                                  {anvil_project, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}           3                                     {anvil_dataset, anvil_project, anvil_file}        25               0.78                                                                                                                                                                                                          [anvil_donor.phenotypic_sex, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_activity.generated_file_id]\n",
       "29         anvil_cmg_broad_orphan_vcgs-white_wes  gtex_ext_3       N            5                                  {anvil_project, anvil_dataset, anvil_donor, anvil_biosample, anvil_file}           3                                     {anvil_dataset, anvil_project, anvil_file}        25               0.78                                                                                                                                                                                                          [anvil_donor.phenotypic_sex, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.donor_age_at_collection_lower_bound, anvil_activity.generated_file_id]\n",
       "30               anvil_cmg_broad_brain_engle_wes  gtex_ext_3       N            4                                               {anvil_biosample, anvil_dataset, anvil_project, anvil_file}           3                                     {anvil_dataset, anvil_project, anvil_file}        24               0.75                                                              [anvil_donor.phenotypic_sex, anvil_donor.donor_id, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_id, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.biosample_id, anvil_biosample.donor_age_at_collection_lower_bound, anvil_activity.used_biosample_id, anvil_activity.activity_id, anvil_activity.generated_file_id]\n",
       "31    anvil_cmg_broad_genitourinary_sinclair_wes  gtex_ext_3       N            4                                               {anvil_biosample, anvil_dataset, anvil_project, anvil_file}           3                                     {anvil_dataset, anvil_project, anvil_file}        24               0.75                                                              [anvil_donor.phenotypic_sex, anvil_donor.donor_id, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_id, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.biosample_id, anvil_biosample.donor_age_at_collection_lower_bound, anvil_activity.used_biosample_id, anvil_activity.activity_id, anvil_activity.generated_file_id]\n",
       "32           anvil_cmg_broad_orphan_jueppner_wes  gtex_ext_3       N            4                                               {anvil_biosample, anvil_dataset, anvil_project, anvil_file}           3                                     {anvil_dataset, anvil_project, anvil_file}        24               0.75                                                              [anvil_donor.phenotypic_sex, anvil_donor.donor_id, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_id, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.biosample_id, anvil_biosample.donor_age_at_collection_lower_bound, anvil_activity.used_biosample_id, anvil_activity.activity_id, anvil_activity.generated_file_id]\n",
       "33              anvil_cmg_broad_orphan_scott_wes  gtex_ext_3       N            4                                               {anvil_biosample, anvil_dataset, anvil_project, anvil_file}           3                                     {anvil_dataset, anvil_project, anvil_file}        24               0.75                                                              [anvil_donor.phenotypic_sex, anvil_donor.donor_id, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_id, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.biosample_id, anvil_biosample.donor_age_at_collection_lower_bound, anvil_activity.used_biosample_id, anvil_activity.activity_id, anvil_activity.generated_file_id]\n",
       "34   anvil_cmg_broad_stillbirth_wilkins-haug_wes  gtex_ext_3       N            4                                               {anvil_biosample, anvil_dataset, anvil_project, anvil_file}           3                                     {anvil_dataset, anvil_project, anvil_file}        24               0.75                                                              [anvil_donor.phenotypic_sex, anvil_donor.donor_id, anvil_biosample.biosample_type, anvil_biosample.donor_age_at_collection_upper_bound, anvil_biosample.donor_id, anvil_biosample.donor_age_at_collection_unit, anvil_biosample.anatomical_site, anvil_biosample.biosample_id, anvil_biosample.donor_age_at_collection_lower_bound, anvil_activity.used_biosample_id, anvil_activity.activity_id, anvil_activity.generated_file_id]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Read the workspace_schemas.csv file into dataframe, clean column names, and convert to dict\n",
    "ws_schema_filepath = ws_bucket + '/ingest_pipeline/resources/mapping_compatibility/workspace_schemas.csv'\n",
    "df = pd.read_csv(ws_schema_filepath)\n",
    "#int_dict = df[df[\"workspace_name\"] == 'anvil_ccdg_asc_ndd_daly_talkowski_aleksic_asd_exome'].to_dict(orient=\"records\")\n",
    "int_dict = df.to_dict(orient=\"records\")\n",
    "\n",
    "# Derived pipeline tables: Should be added to every source workspace schema\n",
    "file_inv_dict = {\"name\": \"file_inventory\",\n",
    "                 \"columns\": [\n",
    "                     {\"name\": \"file_id\"},\n",
    "                     {\"name\": \"name\"},\n",
    "                     {\"name\": \"path\"},\n",
    "                     {\"name\": \"uri\"},\n",
    "                     {\"name\": \"content_type\"},\n",
    "                     {\"name\": \"full_extension\"},\n",
    "                     {\"name\": \"size_in_bytes\"},\n",
    "                     {\"name\": \"crc32c\"},\n",
    "                     {\"name\": \"md5_hash\"},\n",
    "                     {\"name\": \"file_ref\"}\n",
    "                 ]}\n",
    "ws_attr_dict = {\"name\": \"workspace_attributes\",\n",
    "                 \"columns\": [\n",
    "                     {\"name\": \"attribute\"},\n",
    "                     {\"name\": \"value\"}\n",
    "                 ]}\n",
    "\n",
    "# Re-organize workspace dict into TDR-like schema to enable use of existing functions\n",
    "workspace_dict = {}\n",
    "for idx, record in enumerate(int_dict):\n",
    "    if idx == 0:\n",
    "        current_workspace = record[\"workspace_name\"]\n",
    "        previous_workspace = record[\"workspace_name\"]\n",
    "        current_table = record[\"table_name\"]\n",
    "        previous_table = record[\"table_name\"]\n",
    "        current_column = utils.encode_name(record[\"column_name\"])\n",
    "        table_list = []\n",
    "        table_dict = {}\n",
    "        table_dict[\"name\"] = current_table\n",
    "        table_dict[\"columns\"] = []\n",
    "        column_set = set()\n",
    "        column_dict = {}\n",
    "        entity_column = record[\"table_name\"] + \"_id\"\n",
    "        column_dict[\"name\"] = entity_column\n",
    "        column_set.add(entity_column)\n",
    "        table_dict[\"columns\"].append(column_dict)\n",
    "        column_dict = {}\n",
    "        column_dict[\"name\"] = current_column\n",
    "        column_set.add(current_column)\n",
    "        table_dict[\"columns\"].append(column_dict)\n",
    "    else:\n",
    "        current_workspace = record[\"workspace_name\"]\n",
    "        current_table = record[\"table_name\"]\n",
    "        current_column = utils.encode_name(record[\"column_name\"])\n",
    "        if current_workspace != previous_workspace:\n",
    "            table_list.append(table_dict)\n",
    "            table_list.append(file_inv_dict)\n",
    "            table_list.append(ws_attr_dict)\n",
    "            workspace_dict[previous_workspace] = {}\n",
    "            workspace_dict[previous_workspace][\"tables\"] = table_list\n",
    "            table_list = []\n",
    "            table_dict = {}\n",
    "            column_set = set()\n",
    "            table_dict[\"name\"] = current_table\n",
    "            table_dict[\"columns\"] = []\n",
    "            column_dict = {}\n",
    "            entity_column = record[\"table_name\"] + \"_id\"\n",
    "            column_dict[\"name\"] = entity_column\n",
    "            column_set.add(entity_column)\n",
    "            table_dict[\"columns\"].append(column_dict)\n",
    "            column_dict = {}\n",
    "            column_dict[\"name\"] = current_column\n",
    "            column_set.add(current_column)\n",
    "            table_dict[\"columns\"].append(column_dict)\n",
    "        else:\n",
    "            if current_table != previous_table:\n",
    "                for key, value in data_file_refs.items():\n",
    "                    if key.split(\".\")[0] == previous_table:\n",
    "                        for entry in value:\n",
    "                            if utils.encode_name(entry[\"column\"]) in column_set:\n",
    "                                if entry[\"create_new_field\"] == True:\n",
    "                                    column_dict = {}\n",
    "                                    column_dict[\"name\"] = utils.encode_name(entry[\"new_field_name\"])\n",
    "                                    column_set.add(utils.encode_name(entry[\"new_field_name\"]))\n",
    "                                    table_dict[\"columns\"].append(column_dict)\n",
    "                table_list.append(table_dict)\n",
    "                table_dict = {}\n",
    "                column_set = set()\n",
    "                table_dict[\"name\"] = current_table\n",
    "                table_dict[\"columns\"] = []\n",
    "                column_dict = {}\n",
    "                entity_column = record[\"table_name\"] + \"_id\"\n",
    "                column_dict[\"name\"] = entity_column\n",
    "                column_set.add(entity_column)\n",
    "                table_dict[\"columns\"].append(column_dict)\n",
    "                column_dict = {}\n",
    "                column_dict[\"name\"] = current_column\n",
    "                column_set.add(current_column)\n",
    "                table_dict[\"columns\"].append(column_dict)\n",
    "                previous_table = current_table\n",
    "            else:\n",
    "                column_dict = {}\n",
    "                column_dict[\"name\"] = current_column\n",
    "                column_set.add(current_column)\n",
    "                table_dict[\"columns\"].append(column_dict)\n",
    "        previous_workspace = current_workspace\n",
    "        previous_table = current_table\n",
    "    if idx == len(int_dict)-1:\n",
    "        for key, value in data_file_refs.items():\n",
    "            if key.split(\".\")[0] == previous_table:\n",
    "                for entry in value:\n",
    "                    if utils.encode_name(entry[\"column\"]) in column_set:\n",
    "                        if entry[\"create_new_field\"] == True:\n",
    "                            column_dict = {}\n",
    "                            column_dict[\"name\"] = utils.encode_name(entry[\"new_field_name\"])\n",
    "                            column_set.add(utils.encode_name(entry[\"new_field_name\"]))\n",
    "                            table_dict[\"columns\"].append(column_dict)\n",
    "        table_list.append(table_dict)\n",
    "        table_list.append(file_inv_dict)\n",
    "        table_list.append(ws_attr_dict)\n",
    "        workspace_dict[previous_workspace] = {}\n",
    "        workspace_dict[previous_workspace][\"tables\"] = table_list\n",
    "#print(json.dumps(workspace_dict))\n",
    "                              \n",
    "# Read in target schema\n",
    "storage_client = storage.Client()\n",
    "bucket = storage_client.get_bucket(ws_bucket_name)\n",
    "target_schema_dict = {}\n",
    "try:\n",
    "    blob = bucket.blob(f\"ingest_pipeline/mapping/{mapping_target}/mapping_schema_object.json\")\n",
    "    target_schema_dict = json.loads(blob.download_as_string(client=None))\n",
    "except Exception as e:\n",
    "    print(\"Error retrieving target schema for specified mapping_target. Error: {}\".format(e))\n",
    "#print(json.dumps(target_schema_dict))\n",
    "\n",
    "# Loop through mapping specifications for evaluation\n",
    "spec_dict = {}\n",
    "for mapping_target_spec in mapping_target_spec_list:\n",
    "\n",
    "\n",
    "    # Read in mapping specification\n",
    "    storage_client = storage.Client()\n",
    "    bucket = storage_client.get_bucket(ws_bucket_name)\n",
    "    mapping_spec = {}\n",
    "    try:\n",
    "        blob = bucket.blob(f\"ingest_pipeline/mapping/{mapping_target}/{mapping_target_spec}/mapping_specification.json\")\n",
    "        blob_string = blob.download_as_text(client=None)\n",
    "        blob_string = blob_string.replace(\"$DATASET_NAME\", \"Dataset\").replace(\"$PROJECT_NAME\", \"Project\") #UPDATE WITH REAL PARAMETERS\n",
    "        mapping_spec = json.loads(blob_string)\n",
    "    except Exception as e:\n",
    "        print(\"Error retrieving mapping specification for specified mapping_target and mapping_target_spec. Error: {}\".format(e))\n",
    "    #print(json.dumps(mapping_spec))\n",
    "\n",
    "    # Loop through workspaces and evaluate mapping compatibility \n",
    "    detail_dict = {}\n",
    "    for ws_key in workspace_dict:\n",
    "        entity_list = []\n",
    "        for entity in mapping_spec[\"entities\"]:\n",
    "            entity_dict = {}\n",
    "            record_set_list = []\n",
    "            target_table = {}\n",
    "            for table in target_schema_dict[\"tables\"]:\n",
    "                if table[\"name\"] == entity[\"name\"]:\n",
    "                    target_table = table\n",
    "            if target_table:\n",
    "                for record_set in entity[\"record_sets\"]:\n",
    "                    record_set_dict = {}\n",
    "                    record_set_dict[\"record_set\"] = record_set[\"name\"]\n",
    "                    if bmq.validate_record_set(record_set, workspace_dict[ws_key], target_table):\n",
    "                        record_set_dict[\"can_run\"] = True\n",
    "                    else:\n",
    "                        record_set_dict[\"can_run\"] = False\n",
    "                    record_set_dict[\"total_attrs\"] = len(record_set[\"attributes\"])\n",
    "                    valid_attr_count = 0\n",
    "                    invalid_attrs_set = set()\n",
    "                    for attribute in record_set[\"attributes\"]:\n",
    "                        if bmq.validate_attribute(attribute, workspace_dict[ws_key], target_table):\n",
    "                            valid_attr_count += 1\n",
    "                        else:\n",
    "                            invalid_attrs_set.add(target_table[\"name\"] + \".\" + attribute[\"name\"])\n",
    "                    record_set_dict[\"valid_attrs\"] = valid_attr_count\n",
    "                    record_set_dict[\"invalid_attrs_set\"] = list(invalid_attrs_set)\n",
    "                    record_set_list.append(record_set_dict)\n",
    "                entity_dict[entity[\"name\"]] = record_set_list\n",
    "                entity_list.append(entity_dict)\n",
    "        detail_dict[ws_key] = {}\n",
    "        detail_dict[ws_key][\"entities\"] = entity_list\n",
    "    #print(json.dumps(detail_dict))\n",
    "    spec_dict[mapping_target_spec] = {}\n",
    "    spec_dict[mapping_target_spec][\"detail_dict\"] = detail_dict\n",
    "\n",
    "    # Collect target tables and columns not in mapping specification\n",
    "    missing_table_set = set()\n",
    "    missing_column_set = set()\n",
    "    entity_table_list = [val[\"name\"] for val in mapping_spec[\"entities\"]]\n",
    "    entity_column_list = []\n",
    "    for entity in mapping_spec[\"entities\"]:\n",
    "        entity_name = entity[\"name\"]\n",
    "        for record_set in entity[\"record_sets\"]:\n",
    "            for attribute in record_set[\"attributes\"]:\n",
    "                attribute_name = entity_name + \".\" + attribute[\"name\"]\n",
    "                entity_column_list.append(attribute_name)\n",
    "    for table_entry in target_schema_dict[\"tables\"]:\n",
    "        if table_entry[\"name\"] not in entity_table_list:\n",
    "            missing_table_set.add(table_entry[\"name\"])\n",
    "        else:\n",
    "            for column_entry in table_entry[\"columns\"]:\n",
    "                column_name = table_entry[\"name\"] + \".\" + column_entry[\"name\"]\n",
    "                if column_name not in entity_column_list:\n",
    "                    missing_column_set.add(column_name)\n",
    "    spec_dict[mapping_target_spec][\"missing_table_set\"] = missing_table_set\n",
    "    spec_dict[mapping_target_spec][\"missing_column_set\"] = missing_column_set\n",
    "\n",
    "# Summarize mapping compatibility\n",
    "results_list = []\n",
    "for spec_key, spec_val in spec_dict.items():\n",
    "    for ws_key, value in spec_val[\"detail_dict\"].items():\n",
    "        workspace_results_list = []\n",
    "        can_run_set = set()\n",
    "        can_run_fully_set = set()\n",
    "        sum_valid_attrs = 0\n",
    "        sum_total_attrs = 0\n",
    "        invalid_attrs_list = []\n",
    "        for entities in value[\"entities\"]:\n",
    "            max_valid_attrs = 0\n",
    "            max_total_attrs = 0\n",
    "            for key, val in entities.items():\n",
    "                invalid_attrs_set = set(val[0][\"invalid_attrs_set\"])\n",
    "                for record_sets in val:\n",
    "                    if record_sets[\"can_run\"] == True:\n",
    "                        can_run_set.add(key)\n",
    "                    if record_sets[\"total_attrs\"] == record_sets[\"valid_attrs\"]:\n",
    "                        can_run_fully_set.add(key)\n",
    "                    if record_sets[\"valid_attrs\"] > max_valid_attrs:\n",
    "                        max_valid_attrs = record_sets[\"valid_attrs\"]\n",
    "                    if record_sets[\"total_attrs\"] > max_total_attrs:\n",
    "                        max_total_attrs = record_sets[\"total_attrs\"]\n",
    "                    invalid_attrs_set = invalid_attrs_set.union(set(record_sets[\"invalid_attrs_set\"]))\n",
    "                sum_valid_attrs += max_valid_attrs\n",
    "                sum_total_attrs += max_total_attrs\n",
    "                invalid_attrs_list.extend(list(invalid_attrs_set))\n",
    "        percent_valid_attrs = round(sum_valid_attrs/sum_total_attrs,2)\n",
    "        if \"anvil_donor\" in can_run_set and (\"anvil_biosample\" in can_run_set and \"anvil_biosample.donor_id\" not in invalid_attrs_list) and ((\"anvil_activity\" in can_run_set and \"anvil_activity.used_biosample_id\" not in invalid_attrs_list) or (\"anvil_sequencingactivity\" in can_run_set and \"anvil_sequencingactivity.used_biosample_id\" not in invalid_attrs_list)):\n",
    "            compatible = \"Y\"\n",
    "        else:\n",
    "            compatible = \"N\"\n",
    "        workspace_results_list.append(ws_key)\n",
    "        workspace_results_list.append(spec_key)\n",
    "        workspace_results_list.append(compatible)\n",
    "        workspace_results_list.append(len(can_run_set))\n",
    "        workspace_results_list.append(can_run_set)\n",
    "        workspace_results_list.append(len(can_run_fully_set))\n",
    "        workspace_results_list.append(can_run_fully_set)\n",
    "        workspace_results_list.append(sum_valid_attrs)\n",
    "        workspace_results_list.append(percent_valid_attrs)\n",
    "        workspace_results_list.append(invalid_attrs_list)\n",
    "        results_list.append(workspace_results_list)\n",
    "\n",
    "results_df = pd.DataFrame(results_list, columns = ['workspace', 'mapping_spec', 'compatible', 'can_run_count', 'can_run_entities', 'can_fully_run_count', 'can_fully_run_entities', 'cnt_valid_attrs', 'perc_valid_attrs', 'invalid_attr_list'])\n",
    "\n",
    "# Sort results dataframe and write out to tsv\n",
    "destination_dir = \"ingest_pipeline/resources/mapping_compatibility/output\"\n",
    "sorted_df = results_df.sort_values(['mapping_spec', 'compatible', 'can_run_count', 'perc_valid_attrs'], ascending=[True, False, False, False], ignore_index=True)\n",
    "output_file = \"mapping_compatibility_results.tsv\"\n",
    "sorted_df.to_csv(output_file, index=False, sep=\"\\t\")\n",
    "!gsutil cp $output_file $ws_bucket/$destination_dir/ 2> stdout\n",
    "!rm $output_file\n",
    "\n",
    "# Aggregate compatible mapping specs and write out to tsv\n",
    "agg_df = results_df[results_df[\"compatible\"] == \"Y\"].sort_values([\"can_fully_run_count\", \"can_run_count\", \"perc_valid_attrs\", \"mapping_spec\"], ascending=[False, False, False, True]).groupby('workspace').agg(compatible_mapping_specs=('mapping_spec', 'unique')).reset_index()\n",
    "output_file = \"mapping_compatibility_aggregation.tsv\"\n",
    "agg_df.to_csv(output_file, index=False, sep=\"\\t\")\n",
    "!gsutil cp $output_file $ws_bucket/$destination_dir/ 2> stdout\n",
    "!rm $output_file\n",
    "\n",
    "# Output results to the user\n",
    "print(\"------------------------------------------------------------------------------------------------------\")\n",
    "print(\"Mapping Compatibility Results for Mapping Target Specifications:\")\n",
    "print(\"------------------------------------------------------------------------------------------------------\")\n",
    "print(\"Target tables not included in specifications:\")\n",
    "for spec_key, spec_val in spec_dict.items():\n",
    "    print(\"\\t\" + spec_key + \": \" + \", \".join(sorted(list(spec_val[\"missing_table_set\"]))))\n",
    "print(\"\\n\")\n",
    "print(\"Target fields not included in specification:\")\n",
    "for spec_key, spec_val in spec_dict.items():\n",
    "    print(\"\\t\" + spec_key + \": \" + \", \".join(sorted(list(spec_val[\"missing_column_set\"]))))\n",
    "print(\"\\n\")\n",
    "print(\"Workspace evaluation against specifications:\")\n",
    "display(sorted_df)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Utility Scripts"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Pull results for specific workspaces"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "## Print detailed results for specific workspace\n",
    "print(json.dumps(detail_dict[\"anvil_cmg_uwash_ds-hfa\"], indent=2))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "## Print workspace dict for specific workspace\n",
    "print(json.dumps(workspace_dict[\"anvil_gtex_bcm_gru_corsivs\"], indent=2))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Checking workspaces for fileref fields"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "## Checking for fileref fields\n",
    "ws_project = \"anvil-datastorage\"\n",
    "data_file_refs = {}\n",
    "ws_name_list = [\n",
    "\"1000G-high-coverage-2019\"\n",
    "]\n",
    "file_ref_set = set()\n",
    "for ws_name in ws_name_list:\n",
    "    file_ref_list = []\n",
    "    try:\n",
    "        ws_attributes = utils.get_workspace_attributes(ws_project, ws_name)\n",
    "        workspace_bucket = ws_attributes[\"bucketName\"] if ws_attributes.get(\"bucketName\") else \"\" \n",
    "        file_ref_list, data_file_refs, remote_list = utils.find_and_add_fileref_fields(ws_project, ws_name, workspace_bucket, data_file_refs)\n",
    "        for entry in file_ref_list:\n",
    "            file_ref_set.add(entry)\n",
    "    except:\n",
    "        continue\n",
    "print(sorted(list(file_ref_set)))\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Collecting schemas for workspaces"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: center;\">\n",
       "      <th></th>\n",
       "      <th>workspace_name</th>\n",
       "      <th>table_name</th>\n",
       "      <th>column_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>19-disease_description</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>01-subject_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>subject_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>22-age_of_onset</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>07-multiple_datasets</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>20-affected_status</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>11-twin_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>14-ancestry</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>02-prior_testing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>12-proband_relationship</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>25-phenotype_description</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>04-pmid_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>03-project_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>06-dbgap_subject_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>13-sex</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>23-hpo_present</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>21-congenital_status</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>08-family_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>10-maternal_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>09-paternal_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>24-hpo_absent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>17-phenotype_group</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>18-disease_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>26-solve_state</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>16-age_at_last_observation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>05-dbgap_study_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>15-ancestry_detail</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_pair_orientation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>project</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>total_reads</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>md5_path</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_target_bases_2x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_estimated_library_size</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_target_bases_10x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>data_type</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_target_bases_30x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>fold_80_base_penalty</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_pct_exc_dupe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_target_bases_20x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pf_reads_improper_pairs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>read_pair_optical_duplicates</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>mean_target_coverage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>collaborator_sample_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>off_bait_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_target_bases_50x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>median_insert_size</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_percent_duplication</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>bait_set</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_reads_aligned_in_pairs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_selected_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_exc_off_target</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>sample</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>het_snp_q</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-2_pct_exc_dupe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-2_percent_duplication</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_usable_bases_on_bait</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>on_bait_vs_selected</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_exc_overlap</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_off_bait</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>cram_path</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_name</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>width_of_99_percent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>median_target_coverage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>width_of_10_percent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>sequencing_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>near_bait_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pf_hq_error_rate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>crai_path</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>read_pairs_examined</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_exc_baseq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>het_snp_sensitivity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_exc_mapq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>fold_enrichment</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>on_target_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_read_pairs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-2_read_pairs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>mean_read_length</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_usable_bases_on_target</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_target_bases_100x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>strand_balance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-2_estimated_library_size</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>mean_bait_coverage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_exc_dupe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>read_pair_duplicates</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>version</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>on_bait_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_pf_reads_aligned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>target_territory</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>mean_insert_size</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>bait_territory</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_chimeras</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>width_of_50_percent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-2_name</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>93</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-2_pair_orientation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_contamination</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>zero_cvg_targets_pct</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-2_mean_insert_size</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_mean_insert_size</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>collaborator_participant_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sample</td>\n",
       "      <td>02-sample_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sample</td>\n",
       "      <td>04-sequencing_center</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sample</td>\n",
       "      <td>03-dbgap_sample_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sample</td>\n",
       "      <td>sample_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sample</td>\n",
       "      <td>06-tissue_affected_status</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sample</td>\n",
       "      <td>05-sample_source</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>sample</td>\n",
       "      <td>01-subject_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>03-consanguinity_detail</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>07-family_onset</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>family_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>01-family_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>04-pedigree_image</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>111</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>05-pedigree_detail</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>112</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>06-family_history</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>02-consanguinity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>16-sv_type</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>115</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>18-discovery_notes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>116</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>05-inheritance_description</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>01-subject_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>118</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>17-significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>119</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>10-ref</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>120</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>14-transcript</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>06-zygosity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>04-gene_class</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>123</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>11-alt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>124</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>07-variant_genome_build</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>125</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>03-gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>discovery_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>127</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>08-chrom</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>02-sample_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>13-hgvsp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>12-hgvsc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>09-pos</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>132</th>\n",
       "      <td>anvil_cmg_broad_brain_engle_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>15-sv_name</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>19-disease_description</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>134</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>01-subject_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>subject_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>136</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>22-age_of_onset</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>137</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>07-multiple_datasets</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>138</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>20-affected_status</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>11-twin_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>14-ancestry</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>141</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>02-prior_testing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>12-proband_relationship</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>143</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>25-phenotype_description</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>144</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>04-pmid_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>145</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>03-project_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>146</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>06-dbgap_subject_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>147</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>13-sex</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>23-hpo_present</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>149</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>21-congenital_status</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>08-family_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>10-maternal_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>09-paternal_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>153</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>24-hpo_absent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>154</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>17-phenotype_group</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>155</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>18-disease_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>156</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>26-solve_state</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>157</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>16-age_at_last_observation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>158</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>05-dbgap_study_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>159</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>15-ancestry_detail</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>160</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_pair_orientation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>161</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>project</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>total_reads</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>163</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>md5_path</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>164</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_target_bases_2x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_estimated_library_size</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>166</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_target_bases_10x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>167</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>data_type</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>168</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_target_bases_30x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>169</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>fold_80_base_penalty</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_pct_exc_dupe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>171</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_target_bases_20x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>172</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pf_reads_improper_pairs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>173</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>read_pair_optical_duplicates</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>174</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>mean_target_coverage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>175</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>collaborator_sample_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>176</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>off_bait_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>177</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_target_bases_50x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>178</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>median_insert_size</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>179</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_percent_duplication</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>180</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>bait_set</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>181</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_reads_aligned_in_pairs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>182</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_selected_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>183</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_exc_off_target</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>184</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>het_snp_q</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>185</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_usable_bases_on_bait</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>186</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>on_bait_vs_selected</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>187</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_exc_overlap</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>188</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_off_bait</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>189</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>cram_path</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>190</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_name</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>width_of_99_percent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>192</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>median_target_coverage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>width_of_10_percent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>194</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>sequencing_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>195</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>near_bait_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>196</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pf_hq_error_rate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>crai_path</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>198</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>read_pairs_examined</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>199</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_exc_baseq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>het_snp_sensitivity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>201</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_exc_mapq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>202</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>fold_enrichment</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>203</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>on_target_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>204</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_read_pairs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>205</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>mean_read_length</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>206</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_usable_bases_on_target</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>207</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>strand_balance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_target_bases_100x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>209</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>mean_bait_coverage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_exc_dupe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>211</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>read_pair_duplicates</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>212</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>version</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>213</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>on_bait_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>214</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_pf_reads_aligned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>215</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>target_territory</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>216</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>mean_insert_size</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>217</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>bait_territory</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>218</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_chimeras</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>219</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>width_of_50_percent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>220</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_contamination</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>221</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>zero_cvg_targets_pct</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>222</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>sample</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>223</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_mean_insert_size</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>224</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>collaborator_participant_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>225</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sample</td>\n",
       "      <td>02-sample_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>226</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sample</td>\n",
       "      <td>04-sequencing_center</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>227</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sample</td>\n",
       "      <td>03-dbgap_sample_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>228</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sample</td>\n",
       "      <td>sample_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>229</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sample</td>\n",
       "      <td>06-tissue_affected_status</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>230</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sample</td>\n",
       "      <td>05-sample_source</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>231</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>sample</td>\n",
       "      <td>01-subject_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>232</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>03-consanguinity_detail</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>07-family_onset</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>234</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>family_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>235</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>01-family_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>04-pedigree_image</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>237</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>05-pedigree_detail</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>238</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>06-family_history</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>239</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>02-consanguinity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>240</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>16-sv_type</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>241</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>18-discovery_notes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>242</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>05-inheritance_description</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>243</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>01-subject_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>244</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>17-significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>245</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>10-ref</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>246</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>14-transcript</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>247</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>06-zygosity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>248</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>04-gene_class</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>249</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>11-alt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>250</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>07-variant_genome_build</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>251</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>03-gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>252</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>discovery_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>253</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>08-chrom</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>254</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>02-sample_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>255</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>13-hgvsp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>256</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>12-hgvsc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>257</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>09-pos</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>258</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_sinclair_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>15-sv_name</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>259</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>03-consanguinity_detail</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>260</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>07-family_onset</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>261</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>family_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>262</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>01-family_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>263</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>04-pedigree_image</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>264</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>05-pedigree_detail</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>265</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>06-family_history</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>266</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>02-consanguinity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>267</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sample</td>\n",
       "      <td>02-sample_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>268</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sample</td>\n",
       "      <td>04-sequencing_center</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>269</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sample</td>\n",
       "      <td>03-dbgap_sample_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>270</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sample</td>\n",
       "      <td>sample_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sample</td>\n",
       "      <td>06-tissue_affected_status</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>272</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sample</td>\n",
       "      <td>05-sample_source</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>273</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sample</td>\n",
       "      <td>01-subject_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>274</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_pair_orientation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>275</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>project</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>276</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>total_reads</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>277</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>md5_path</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>278</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_target_bases_2x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>279</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_estimated_library_size</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>280</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_target_bases_10x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>281</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>data_type</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>282</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_target_bases_30x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>283</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>fold_80_base_penalty</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>284</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_pct_exc_dupe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>285</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_target_bases_20x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>286</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pf_reads_improper_pairs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>287</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>read_pair_optical_duplicates</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>288</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>mean_target_coverage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>289</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>collaborator_sample_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>290</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>off_bait_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>291</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_target_bases_50x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>292</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>median_insert_size</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>293</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_percent_duplication</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>294</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>bait_set</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>295</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_reads_aligned_in_pairs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>296</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_selected_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>297</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_exc_off_target</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>298</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>het_snp_q</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>299</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_usable_bases_on_bait</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>300</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>on_bait_vs_selected</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>301</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_exc_overlap</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>302</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_off_bait</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>cram_path</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>304</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_name</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>305</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>width_of_99_percent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>median_target_coverage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>307</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>width_of_10_percent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>308</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>sequencing_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>309</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>near_bait_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>310</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pf_hq_error_rate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>311</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>crai_path</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>312</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>read_pairs_examined</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>313</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_exc_baseq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>314</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>het_snp_sensitivity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>315</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_exc_mapq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>fold_enrichment</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>317</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>on_target_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>318</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_read_pairs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>319</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>mean_read_length</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>320</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_usable_bases_on_target</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>321</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>strand_balance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>322</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_target_bases_100x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>323</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>mean_bait_coverage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>324</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_exc_dupe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>325</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>read_pair_duplicates</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>326</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>version</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>327</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>on_bait_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_pf_reads_aligned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>329</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>target_territory</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>330</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>mean_insert_size</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>331</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>bait_territory</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>332</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_chimeras</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>width_of_50_percent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_contamination</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>335</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>zero_cvg_targets_pct</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>336</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>sample</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>337</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_mean_insert_size</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>338</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>collaborator_participant_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>339</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>19-disease_description</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>01-subject_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>341</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>subject_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>342</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>22-age_of_onset</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>343</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>07-multiple_datasets</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>344</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>20-affected_status</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>11-twin_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>346</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>14-ancestry</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>347</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>02-prior_testing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>348</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>12-proband_relationship</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>349</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>25-phenotype_description</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>350</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>04-pmid_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>351</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>03-project_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>352</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>06-dbgap_subject_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>353</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>13-sex</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>354</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>23-hpo_present</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>355</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>21-congenital_status</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>356</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>08-family_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>357</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>10-maternal_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>358</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>09-paternal_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>359</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>24-hpo_absent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>360</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>17-phenotype_group</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>361</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>18-disease_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>362</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>26-solve_state</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>363</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>16-age_at_last_observation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>364</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>05-dbgap_study_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>365</th>\n",
       "      <td>anvil_cmg_broad_orphan_jueppner_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>15-ancestry_detail</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>366</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>participant</td>\n",
       "      <td>collaborator_participant_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>367</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>participant</td>\n",
       "      <td>gender</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>368</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>participant</td>\n",
       "      <td>participant_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>369</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>19-disease_description</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>370</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>01-subject_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>371</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>subject_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>372</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>22-age_of_onset</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>373</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>07-multiple_datasets</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>374</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>20-affected_status</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>375</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>11-twin_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>376</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>14-ancestry</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>377</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>02-prior_testing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>378</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>12-proband_relationship</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>379</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>25-phenotype_description</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>380</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>04-pmid_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>381</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>03-project_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>382</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>06-dbgap_subject_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>383</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>13-sex</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>384</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>23-hpo_present</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>385</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>21-congenital_status</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>386</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>08-family_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>387</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>10-maternal_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>388</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>09-paternal_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>389</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>24-hpo_absent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>390</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>17-phenotype_group</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>391</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>18-disease_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>392</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>26-solve_state</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>393</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>16-age_at_last_observation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>394</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>05-dbgap_study_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>15-ancestry_detail</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>396</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_het_snp_sensitivity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>397</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>at_dropout</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>398</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_unpaired_reads_examined</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>399</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>is_primary</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>400</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_pct_target_bases_20x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>401</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_off_bait_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>402</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>novel_titv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>403</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>num_in_db_snp_multiallelic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>404</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>rna_pct_intronic_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>405</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>wgs_pct_exc_unpaired</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>406</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_target_bases_10x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>407</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>rna_ribosomal_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>408</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_pf_bases_aligned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>409</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>novel_indels</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>410</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>mean_target_coverage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>411</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pf_hq_median_mismatches</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>off_bait_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>413</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hom_as_het</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>414</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>rna_pct_r2_transcript_strand_reads</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>415</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>rna_pct_r1_transcript_strand_reads</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>416</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pf_indel_rate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>417</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_reads_aligned_in_pairs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>418</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_selected_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>419</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>rna_pf_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_target_territory</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>421</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>num_in_db_snp_complex_indels</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>422</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>median_absolute_deviation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>423</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_read_pair_duplicates</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>424</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>rna_incorrect_strand_reads</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>425</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>total_complex_indels</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>426</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>wgs_pct_50x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>427</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>width_of_90_percent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>428</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>wgs_pct_20x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>429</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_pct_pf_reads</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>430</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_hs_penalty_20x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>431</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>rna_utr_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>432</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_pct_pf_uq_reads</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>433</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_on_bait_vs_selected</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>434</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>rna_ignored_reads</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>wgs_pct_25x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>436</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_bait_design_efficiency</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>437</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>het_snp_sensitivity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>438</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_exc_mapq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>439</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>fold_enrichment</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>440</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>mean_read_length</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>441</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>width_of_30_percent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>442</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>gc_nc_40_59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>443</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_median_target_coverage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>444</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>width_of_40_percent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>445</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_exc_dupe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>446</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>gc_nc_80_100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>447</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>rna_coding_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>448</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>rna_pct_ribosomal_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>449</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>bait_territory</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>450</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hom_as_other_hom</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>451</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>filtered_snps</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>452</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_pct_target_bases_2x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>453</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_pf_unique_reads</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>454</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_mean_insert_size</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>455</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pf_hq_error_rate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>456</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>window_size</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>457</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_pair_orientation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>458</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>rna_correct_strand_reads</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>459</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>rna_pct_mrna_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>460</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_pct_usable_bases_on_bait</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>461</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>wgs_pct_70x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>462</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-2_unmapped_reads</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>463</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>width_of_80_percent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>464</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_estimated_library_size</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>465</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_target_bases_30x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>466</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>total_indels</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>467</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>wgs_pct_30x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>468</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>dbsnp_titv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>469</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_hs_penalty_30x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>470</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>collaborator_sample_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>471</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>read_pairs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>472</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>rna_pf_aligned_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>473</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>wgs_pct_5x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>474</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>475</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>bait_set</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>476</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>rna_pct_utr_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>477</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>wgs_pct_15x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>478</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>wgs_pct_exc_baseq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>479</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_gc_dropout</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>480</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-2_percent_duplication</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>481</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>wgs_het_snp_sensitivity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>482</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>num_snps</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>483</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>max_insert_size</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>484</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>width_of_99_percent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>485</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>haplotypes_confidently_matchin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>486</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>median_target_coverage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>487</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>width_of_10_percent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>488</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>sequencing_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>489</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-2_secondary_or_supplementary_rds</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>490</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>near_bait_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>491</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>wgs_sd_coverage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>492</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_exc_baseq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>493</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_zero_cvg_targets_pct</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>494</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_read_pairs_examined</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>495</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_pf_reads</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>496</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_pct_exc_overlap</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>497</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_read_pairs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>498</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>rna_num_r2_transcript_strand_reads</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_target_bases_100x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>500</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>wgs_het_snp_q</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>501</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_pct_target_bases_30x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>novel_ins_del_ratio</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>503</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_hs_penalty_40x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>504</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pf_reads_aligned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>505</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_pf_reads_aligned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>506</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>rna_intergenic_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>507</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>gc_nc_20_39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>508</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_chimeras</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>509</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>haplotypes_confidently_checked</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>510</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pf_hq_aligned_q20_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>511</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>zero_cvg_targets_pct</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>512</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>num_in_db_snp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>513</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_near_bait_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>514</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_at_dropout</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>515</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>participant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>516</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>wgs_pct_exc_dupe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>517</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>wgs_pct_10x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>518</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_fold_80_base_penalty</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>519</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>wgs_pct_60x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>520</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pf_reads</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>521</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>wgs_mad_coverage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>md5_path</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>523</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_read_pair_optical_duplicates</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>524</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>wgs_pct_exc_total</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>525</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_pct_target_bases_50x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>526</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pf_hq_aligned_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>527</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>rna_num_unexplained_reads</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>528</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pdo</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>529</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_pct_exc_dupe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>530</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>het_as_hom</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>531</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_genome_size</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>532</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_mean_target_coverage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>533</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>crai_or_bai_path</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>534</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>mean_depth</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>535</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>ll_random_sample</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>536</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_pf_reads</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>537</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_hs_penalty_10x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>538</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_target_bases_20x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>539</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_target_bases_50x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>540</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>wgs_pct_100x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>541</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>reads_aligned_in_pairs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>542</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>rna_pct_correct_strand_reads</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>543</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>aligned_reads</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>544</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_het_snp_q</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>545</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_percent_duplication</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>546</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_pct_exc_baseq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>547</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_dbsnp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>548</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_hs_penalty_50x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>549</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>total_clusters</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>550</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>wgs_median_coverage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>551</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>rna_median_cv_coverage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>552</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>num_singletons</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>553</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_usable_bases_on_bait</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>554</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>snp_reference_bias</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>555</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>on_bait_vs_selected</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>556</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>num_in_db_snp_indels</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>557</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>rna_median_5prime_to_3prime_bias</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>558</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_min_target_coverage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>559</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>lod_expected_sample</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>560</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>cram_path</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>561</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>total_multiallelic_snps</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>562</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_pct_exc_dupe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>563</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>ll_predicted_contam</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>564</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>reference_sequence_name</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>565</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>ll_no_contam</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>566</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pf_noise_reads</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>567</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>rna_intronic_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>568</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>gc_dropout</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>569</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pf_aligned_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>570</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>rna_num_r1_transcript_strand_reads</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>571</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>release_date</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>572</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_usable_bases_on_target</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>573</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>num_reads</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>574</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>width_of_60_percent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>575</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>sequencing_project</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>576</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>rna_pct_coding_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>577</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>ll_expected_sample</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>578</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>mean_bait_coverage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>579</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>haplotypes_with_genotypes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>580</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>lc-set</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>581</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_chimerism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>582</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>on_bait_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>583</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>research_project</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>584</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>rna_median_5prime_bias</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>585</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>version</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>586</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>wgs_pct_exc_overlap</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>587</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>sample_alias</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>588</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_contamination</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>589</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>read_groups</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>590</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_adapter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>591</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>sample</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>592</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_max_target_coverage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>593</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>wgs_pct_1x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>594</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_bait_territory</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>595</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_pct_target_bases_10x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>596</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>collaborator_participant_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>597</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pf_hq_aligned_reads</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>598</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>project</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>599</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>total_reads</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>600</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>wgs_pct_exc_mapq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>601</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>total_snps</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>602</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_on_target_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>603</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_target_bases_2x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>604</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pf_mismatch_rate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>605</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>data_type</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>606</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_fold_enrichment</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>607</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>fold_80_base_penalty</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>608</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_on_bait_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>609</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_pct_target_bases_100x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>610</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_hs_library_size</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>611</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>wgs_genome_territory</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>612</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_mean_bait_coverage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>613</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_dbsnp_indels</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>614</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pair_orientation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>615</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-2_read_pairs_examined</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>616</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>median_insert_size</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>617</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_pct_target_bases_1x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>618</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>individual_alias</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>619</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_unpaired_read_duplicates</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>620</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>read_group</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>621</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>width_of_20_percent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>622</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>gc_nc_0_19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>623</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_exc_off_target</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>624</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>sample_type</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>625</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>het_snp_q</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>626</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_pct_pf_uq_reads_aligned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>627</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_pct_exc_mapq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>628</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_pct_usable_bases_on_target</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>629</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_exc_overlap</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>630</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>novel_snps</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>631</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_pct_target_bases_40x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>632</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_pct_exc_off_target</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>633</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_off_bait</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>634</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>standard_deviation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>635</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>wgs_pct_80x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>636</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_name</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>637</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-2_unpaired_read_duplicates</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>638</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>dbsnp_ins_del_ratio</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>639</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_unmapped_reads</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>640</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_pct_selected_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>641</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_total_reads</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>642</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-2_read_pair_optical_duplicates</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>643</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>wgs_pct_40x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>644</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>crai_path</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>645</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-2_read_pair_duplicates</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>646</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>wgs_pct_90x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>647</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>strand_balance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>648</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>on_target_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>649</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_pf_uq_bases_aligned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>650</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>wgs_mean_coverage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>651</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_hs_penalty_100x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>652</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-2_estimated_library_size</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>653</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>bad_cycles</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>654</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>min_insert_size</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>655</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>wgs_pct_exc_capped</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>656</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>width_of_70_percent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>657</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-2_unpaired_reads_examined</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>658</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>target_territory</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>659</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_pf_uq_reads_aligned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>660</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>mean_insert_size</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>661</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>filtered_indels</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>662</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>gc_nc_60_79</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>663</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>hs_pct_off_bait</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>664</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_secondary_or_supplementary_rds</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>665</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-2_name</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>666</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>rna_pct_intergenic_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>667</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>rna_pct_usable_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>668</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>cram_or_bam_path</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>669</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>width_of_50_percent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>670</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>rna_median_3prime_bias</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>671</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sample</td>\n",
       "      <td>02-sample_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>672</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sample</td>\n",
       "      <td>04-sequencing_center</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>673</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sample</td>\n",
       "      <td>03-dbgap_sample_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>674</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sample</td>\n",
       "      <td>sample_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>675</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sample</td>\n",
       "      <td>06-tissue_affected_status</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>676</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sample</td>\n",
       "      <td>05-sample_source</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>677</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>sample</td>\n",
       "      <td>01-subject_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>678</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>03-consanguinity_detail</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>679</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>07-family_onset</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>680</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>family_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>681</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>01-family_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>682</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>04-pedigree_image</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>683</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>05-pedigree_detail</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>684</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>06-family_history</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>685</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>02-consanguinity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>686</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>16-sv_type</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>687</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>18-discovery_notes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>688</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>05-inheritance_description</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>689</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>01-subject_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>690</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>17-significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>691</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>10-ref</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>692</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>14-transcript</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>693</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>06-zygosity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>694</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>04-gene_class</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>695</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>11-alt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>696</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>07-variant_genome_build</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>697</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>03-gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>698</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>discovery_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>699</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>08-chrom</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>700</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>02-sample_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>701</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>13-hgvsp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>702</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>12-hgvsc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>703</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>09-pos</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>704</th>\n",
       "      <td>anvil_cmg_broad_genitourinary_hirschhorn_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>15-sv_name</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>705</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>19-disease_description</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>706</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>01-subject_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>707</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>subject_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>708</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>22-age_of_onset</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>709</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>07-multiple_datasets</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>710</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>20-affected_status</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>711</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>11-twin_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>712</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>14-ancestry</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>713</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>02-prior_testing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>714</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>12-proband_relationship</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>715</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>25-phenotype_description</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>716</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>04-pmid_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>717</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>03-project_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>718</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>06-dbgap_subject_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>719</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>13-sex</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>720</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>23-hpo_present</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>721</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>21-congenital_status</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>722</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>08-family_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>723</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>10-maternal_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>724</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>09-paternal_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>725</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>24-hpo_absent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>726</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>17-phenotype_group</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>727</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>18-disease_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>728</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>26-solve_state</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>729</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>16-age_at_last_observation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>730</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>05-dbgap_study_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>731</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>15-ancestry_detail</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>732</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_pair_orientation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>733</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>project</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>734</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>total_reads</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>735</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>md5_path</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>736</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_target_bases_2x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>737</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_estimated_library_size</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>738</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_target_bases_10x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>739</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>data_type</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>740</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_target_bases_30x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>741</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>fold_80_base_penalty</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>742</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_pct_exc_dupe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>743</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_target_bases_20x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>744</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pf_reads_improper_pairs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>745</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>read_pair_optical_duplicates</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>746</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>mean_target_coverage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>747</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>collaborator_sample_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>748</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>off_bait_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>749</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_target_bases_50x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>750</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>median_insert_size</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>751</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_percent_duplication</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>752</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>bait_set</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>753</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_reads_aligned_in_pairs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>754</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_selected_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>755</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_exc_off_target</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>756</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>sample</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>757</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>het_snp_q</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>758</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-2_pct_exc_dupe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>759</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-2_percent_duplication</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>760</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_usable_bases_on_bait</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>761</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>on_bait_vs_selected</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>762</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_exc_overlap</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>763</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_off_bait</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>764</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>cram_path</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>765</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_name</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>766</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>width_of_99_percent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>767</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>median_target_coverage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>768</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>width_of_10_percent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>769</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>sequencing_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>770</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>near_bait_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>771</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pf_hq_error_rate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>772</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>crai_path</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>773</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>read_pairs_examined</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>774</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_exc_baseq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>775</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>het_snp_sensitivity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>776</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_exc_mapq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>777</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>fold_enrichment</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>778</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>on_target_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>779</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_read_pairs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>780</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-2_read_pairs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>781</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>mean_read_length</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>782</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_usable_bases_on_target</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>783</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_target_bases_100x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>784</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>strand_balance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>785</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-2_estimated_library_size</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>786</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>mean_bait_coverage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>787</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_exc_dupe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>788</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>read_pair_duplicates</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>789</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>version</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>790</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>on_bait_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>791</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_pf_reads_aligned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>792</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>target_territory</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>793</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>mean_insert_size</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>794</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>bait_territory</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>795</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_chimeras</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>796</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>width_of_50_percent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>797</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-2_name</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>798</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-2_pair_orientation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>799</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_contamination</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>800</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>zero_cvg_targets_pct</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>801</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-2_mean_insert_size</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>802</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_mean_insert_size</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>803</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>collaborator_participant_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>804</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sample</td>\n",
       "      <td>02-sample_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>805</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sample</td>\n",
       "      <td>04-sequencing_center</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>806</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sample</td>\n",
       "      <td>03-dbgap_sample_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>807</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sample</td>\n",
       "      <td>sample_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>808</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sample</td>\n",
       "      <td>06-tissue_affected_status</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>809</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sample</td>\n",
       "      <td>05-sample_source</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>810</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>sample</td>\n",
       "      <td>01-subject_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>811</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>03-consanguinity_detail</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>812</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>07-family_onset</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>813</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>family_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>814</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>01-family_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>815</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>04-pedigree_image</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>816</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>05-pedigree_detail</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>817</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>06-family_history</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>818</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>02-consanguinity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>819</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>16-sv_type</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>820</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>18-discovery_notes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>821</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>05-inheritance_description</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>822</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>01-subject_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>823</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>17-significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>824</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>10-ref</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>825</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>14-transcript</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>826</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>06-zygosity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>827</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>04-gene_class</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>828</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>11-alt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>829</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>07-variant_genome_build</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>830</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>03-gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>831</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>discovery_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>832</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>08-chrom</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>833</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>02-sample_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>13-hgvsp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>835</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>12-hgvsc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>836</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>09-pos</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>837</th>\n",
       "      <td>anvil_cmg_broad_orphan_scott_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>15-sv_name</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>838</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>participant</td>\n",
       "      <td>collaborator_participant_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>839</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>participant</td>\n",
       "      <td>gender</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>840</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>participant</td>\n",
       "      <td>participant_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>841</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>19-disease_description</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>842</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>01-subject_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>843</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>subject_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>844</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>22-age_of_onset</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>845</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>07-multiple_datasets</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>846</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>20-affected_status</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>847</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>11-twin_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>848</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>14-ancestry</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>849</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>02-prior_testing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>850</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>12-proband_relationship</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>851</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>25-phenotype_description</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>852</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>04-pmid_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>853</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>03-project_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>854</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>06-dbgap_subject_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>855</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>13-sex</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>856</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>23-hpo_present</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>857</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>21-congenital_status</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>858</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>08-family_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>859</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>10-maternal_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>860</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>09-paternal_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>861</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>24-hpo_absent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>862</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>17-phenotype_group</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>863</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>18-disease_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>864</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>26-solve_state</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>865</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>16-age_at_last_observation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>866</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>05-dbgap_study_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>867</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>15-ancestry_detail</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>868</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_target_bases_10x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>869</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>20x_rate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>870</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>mean_target_coverage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>871</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>off_bait_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>872</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_selected_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>873</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-2_pct_exc_dupe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>874</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pdo</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>875</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>exc_unpaired_rate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>876</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pf_reads_rate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>877</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>adapter_rate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>878</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>het_snp_sensitivity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>879</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>fold_enrichment</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>880</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_exc_mapq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>881</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>mean_read_length</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>882</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_exc_dupe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>883</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>bait_territory</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>884</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-2_mean_insert_size</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>885</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_mean_insert_size</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>886</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_pair_orientation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>887</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_estimated_library_size</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>888</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_target_bases_30x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>889</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>genome_territory</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>890</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>collaborator_sample_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>891</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>bait_set</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>892</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-2_percent_duplication</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>893</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>median_target_coverage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>894</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>sequencing_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>895</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>near_bait_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>896</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_exc_baseq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>897</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_read_pairs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>898</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_target_bases_100x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>899</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pf_reads_aligned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>900</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>reads_aligned_in_pairs_rate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>901</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_chimeras</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>902</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pf_hq_aligned_q20_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>903</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>zero_cvg_targets_pct</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>904</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pf_reads</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>905</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>mean_coverage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>906</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>md5_path</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>907</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>exc_baseq_rate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>908</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pf_hq_aligned_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>909</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_pct_exc_dupe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>910</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_target_bases_20x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>911</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_target_bases_50x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>912</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>reads_aligned_in_pairs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>913</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_percent_duplication</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>914</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>10x_rate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>915</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_usable_bases_on_bait</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>916</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>exc_overlap_rate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>917</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>on_bait_vs_selected</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>918</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>cram_path</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>919</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>reference_sequence_name</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>920</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pf_noise_reads</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>921</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pf_aligned_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>922</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>release_date</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>923</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_usable_bases_on_target</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>924</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>mean_bait_coverage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>925</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>version</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>926</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>on_bait_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>927</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>sample</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>928</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>collaborator_participant_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>929</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>contamination_rate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>930</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pf_hq_aligned_reads</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>931</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>project</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>932</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>total_reads</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>933</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>exc_dupe_rate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>934</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_target_bases_2x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>935</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pf_mismatch_rate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>936</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>data_type</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>937</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>fold_80_base_penalty</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>938</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>exc_total_rate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>939</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_exc_off_target</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>940</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>het_snp_q</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>941</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pf_reads_aligned_rate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>942</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_exc_overlap</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>943</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_off_bait</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>944</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_name</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>945</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>crai_path</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>946</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>on_target_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>947</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>chimera_rate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>948</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-2_estimated_library_size</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>949</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>exc_mapq_rate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>950</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>target_territory</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>951</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>root_sample_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>952</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-2_name</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>953</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sample</td>\n",
       "      <td>02-sample_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>954</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sample</td>\n",
       "      <td>04-sequencing_center</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>955</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sample</td>\n",
       "      <td>03-dbgap_sample_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>956</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sample</td>\n",
       "      <td>sample_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>957</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sample</td>\n",
       "      <td>06-tissue_affected_status</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>958</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sample</td>\n",
       "      <td>05-sample_source</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>959</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>sample</td>\n",
       "      <td>01-subject_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>960</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>03-consanguinity_detail</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>961</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>07-family_onset</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>962</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>family_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>963</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>01-family_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>964</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>04-pedigree_image</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>965</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>05-pedigree_detail</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>966</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>06-family_history</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>967</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>02-consanguinity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>968</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>16-sv_type</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>969</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>18-discovery_notes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>970</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>05-inheritance_description</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>971</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>01-subject_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>972</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>17-significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>973</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>10-ref</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>974</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>14-transcript</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>975</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>06-zygosity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>976</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>04-gene_class</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>977</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>11-alt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>978</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>07-variant_genome_build</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>979</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>03-gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>980</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>discovery_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>981</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>08-chrom</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>982</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>02-sample_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>983</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>13-hgvsp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>984</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>12-hgvsc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>985</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>09-pos</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>986</th>\n",
       "      <td>anvil_cmg_broad_orphan_vcgs-white_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>15-sv_name</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>987</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>19-disease_description</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>988</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>01-subject_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>989</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>subject_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>990</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>22-age_of_onset</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>991</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>07-multiple_datasets</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>992</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>20-affected_status</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>993</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>11-twin_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>994</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>14-ancestry</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>995</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>02-prior_testing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>996</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>12-proband_relationship</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>997</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>25-phenotype_description</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>998</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>04-pmid_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>999</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>03-project_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1000</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>06-dbgap_subject_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1001</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>13-sex</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1002</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>23-hpo_present</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1003</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>21-congenital_status</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1004</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>08-family_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1005</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>10-maternal_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1006</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>09-paternal_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1007</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>24-hpo_absent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1008</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>17-phenotype_group</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1009</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>18-disease_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1010</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>26-solve_state</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1011</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>16-age_at_last_observation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1012</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>05-dbgap_study_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1013</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>subject</td>\n",
       "      <td>15-ancestry_detail</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1014</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_pair_orientation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1015</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>project</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1016</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>total_reads</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1017</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>md5_path</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1018</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_target_bases_2x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1019</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_estimated_library_size</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1020</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_target_bases_10x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1021</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>data_type</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1022</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_target_bases_30x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1023</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>fold_80_base_penalty</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1024</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_pct_exc_dupe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1025</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_target_bases_20x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1026</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pf_reads_improper_pairs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1027</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>read_pair_optical_duplicates</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1028</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>mean_target_coverage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1029</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>collaborator_sample_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1030</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>off_bait_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1031</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_target_bases_50x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1032</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>median_insert_size</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1033</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_percent_duplication</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1034</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>bait_set</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1035</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_reads_aligned_in_pairs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1036</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_selected_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1037</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_exc_off_target</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1038</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>het_snp_q</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1039</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_usable_bases_on_bait</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1040</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>on_bait_vs_selected</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1041</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_exc_overlap</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1042</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_off_bait</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1043</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>cram_path</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1044</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_name</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1045</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>width_of_99_percent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1046</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>median_target_coverage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1047</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>width_of_10_percent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1048</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>sequencing_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1049</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>near_bait_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1050</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pf_hq_error_rate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1051</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>crai_path</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1052</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>read_pairs_examined</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1053</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_exc_baseq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1054</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>het_snp_sensitivity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1055</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_exc_mapq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1056</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>fold_enrichment</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1057</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>on_target_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1058</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_read_pairs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1059</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>mean_read_length</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1060</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_usable_bases_on_target</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1061</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>strand_balance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1062</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_target_bases_100x</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1063</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>mean_bait_coverage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1064</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_exc_dupe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1065</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>read_pair_duplicates</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1066</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>version</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1067</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>on_bait_bases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1068</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_pf_reads_aligned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1069</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>target_territory</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1070</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>mean_insert_size</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1071</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>bait_territory</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1072</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_chimeras</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1073</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>width_of_50_percent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1074</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>pct_contamination</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1075</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>zero_cvg_targets_pct</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1076</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>sample</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1077</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>library-1_mean_insert_size</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1078</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sequencing</td>\n",
       "      <td>collaborator_participant_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1079</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sample</td>\n",
       "      <td>02-sample_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1080</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sample</td>\n",
       "      <td>04-sequencing_center</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1081</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sample</td>\n",
       "      <td>03-dbgap_sample_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1082</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sample</td>\n",
       "      <td>sample_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1083</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sample</td>\n",
       "      <td>06-tissue_affected_status</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1084</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sample</td>\n",
       "      <td>05-sample_source</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1085</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>sample</td>\n",
       "      <td>01-subject_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1086</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>03-consanguinity_detail</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1087</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>07-family_onset</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1088</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>family_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1089</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>01-family_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1090</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>04-pedigree_image</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1091</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>05-pedigree_detail</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1092</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>06-family_history</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1093</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>family</td>\n",
       "      <td>02-consanguinity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1094</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>16-sv_type</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1095</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>18-discovery_notes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1096</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>05-inheritance_description</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1097</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>01-subject_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1098</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>17-significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1099</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>10-ref</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1100</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>14-transcript</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1101</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>06-zygosity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1102</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>04-gene_class</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1103</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>11-alt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1104</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>07-variant_genome_build</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1105</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>03-gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1106</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>discovery_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1107</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>08-chrom</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1108</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>02-sample_id</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1109</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>13-hgvsp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1110</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>12-hgvsc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1111</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>09-pos</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1112</th>\n",
       "      <td>anvil_cmg_broad_stillbirth_wilkins-haug_wes</td>\n",
       "      <td>discovery</td>\n",
       "      <td>15-sv_name</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     workspace_name                 table_name                 column_name               \n",
       "0                  anvil_cmg_broad_brain_engle_wes      subject                    19-disease_description\n",
       "1                  anvil_cmg_broad_brain_engle_wes      subject                             01-subject_id\n",
       "2                  anvil_cmg_broad_brain_engle_wes      subject                                subject_id\n",
       "3                  anvil_cmg_broad_brain_engle_wes      subject                           22-age_of_onset\n",
       "4                  anvil_cmg_broad_brain_engle_wes      subject                      07-multiple_datasets\n",
       "5                  anvil_cmg_broad_brain_engle_wes      subject                        20-affected_status\n",
       "6                  anvil_cmg_broad_brain_engle_wes      subject                                11-twin_id\n",
       "7                  anvil_cmg_broad_brain_engle_wes      subject                               14-ancestry\n",
       "8                  anvil_cmg_broad_brain_engle_wes      subject                          02-prior_testing\n",
       "9                  anvil_cmg_broad_brain_engle_wes      subject                   12-proband_relationship\n",
       "10                 anvil_cmg_broad_brain_engle_wes      subject                  25-phenotype_description\n",
       "11                 anvil_cmg_broad_brain_engle_wes      subject                                04-pmid_id\n",
       "12                 anvil_cmg_broad_brain_engle_wes      subject                             03-project_id\n",
       "13                 anvil_cmg_broad_brain_engle_wes      subject                       06-dbgap_subject_id\n",
       "14                 anvil_cmg_broad_brain_engle_wes      subject                                    13-sex\n",
       "15                 anvil_cmg_broad_brain_engle_wes      subject                            23-hpo_present\n",
       "16                 anvil_cmg_broad_brain_engle_wes      subject                      21-congenital_status\n",
       "17                 anvil_cmg_broad_brain_engle_wes      subject                              08-family_id\n",
       "18                 anvil_cmg_broad_brain_engle_wes      subject                            10-maternal_id\n",
       "19                 anvil_cmg_broad_brain_engle_wes      subject                            09-paternal_id\n",
       "20                 anvil_cmg_broad_brain_engle_wes      subject                             24-hpo_absent\n",
       "21                 anvil_cmg_broad_brain_engle_wes      subject                        17-phenotype_group\n",
       "22                 anvil_cmg_broad_brain_engle_wes      subject                             18-disease_id\n",
       "23                 anvil_cmg_broad_brain_engle_wes      subject                            26-solve_state\n",
       "24                 anvil_cmg_broad_brain_engle_wes      subject                16-age_at_last_observation\n",
       "25                 anvil_cmg_broad_brain_engle_wes      subject                         05-dbgap_study_id\n",
       "26                 anvil_cmg_broad_brain_engle_wes      subject                        15-ancestry_detail\n",
       "27                 anvil_cmg_broad_brain_engle_wes   sequencing                library-1_pair_orientation\n",
       "28                 anvil_cmg_broad_brain_engle_wes   sequencing                                   project\n",
       "29                 anvil_cmg_broad_brain_engle_wes   sequencing                               total_reads\n",
       "30                 anvil_cmg_broad_brain_engle_wes   sequencing                                  md5_path\n",
       "31                 anvil_cmg_broad_brain_engle_wes   sequencing                       pct_target_bases_2x\n",
       "32                 anvil_cmg_broad_brain_engle_wes   sequencing          library-1_estimated_library_size\n",
       "33                 anvil_cmg_broad_brain_engle_wes   sequencing                      pct_target_bases_10x\n",
       "34                 anvil_cmg_broad_brain_engle_wes   sequencing                                 data_type\n",
       "35                 anvil_cmg_broad_brain_engle_wes   sequencing                      pct_target_bases_30x\n",
       "36                 anvil_cmg_broad_brain_engle_wes   sequencing                      fold_80_base_penalty\n",
       "37                 anvil_cmg_broad_brain_engle_wes   sequencing                    library-1_pct_exc_dupe\n",
       "38                 anvil_cmg_broad_brain_engle_wes   sequencing                      pct_target_bases_20x\n",
       "39                 anvil_cmg_broad_brain_engle_wes   sequencing                   pf_reads_improper_pairs\n",
       "40                 anvil_cmg_broad_brain_engle_wes   sequencing              read_pair_optical_duplicates\n",
       "41                 anvil_cmg_broad_brain_engle_wes   sequencing                      mean_target_coverage\n",
       "42                 anvil_cmg_broad_brain_engle_wes   sequencing                    collaborator_sample_id\n",
       "43                 anvil_cmg_broad_brain_engle_wes   sequencing                            off_bait_bases\n",
       "44                 anvil_cmg_broad_brain_engle_wes   sequencing                      pct_target_bases_50x\n",
       "45                 anvil_cmg_broad_brain_engle_wes   sequencing                        median_insert_size\n",
       "46                 anvil_cmg_broad_brain_engle_wes   sequencing             library-1_percent_duplication\n",
       "47                 anvil_cmg_broad_brain_engle_wes   sequencing                                  bait_set\n",
       "48                 anvil_cmg_broad_brain_engle_wes   sequencing                pct_reads_aligned_in_pairs\n",
       "49                 anvil_cmg_broad_brain_engle_wes   sequencing                        pct_selected_bases\n",
       "50                 anvil_cmg_broad_brain_engle_wes   sequencing                        pct_exc_off_target\n",
       "51                 anvil_cmg_broad_brain_engle_wes   sequencing                                    sample\n",
       "52                 anvil_cmg_broad_brain_engle_wes   sequencing                                 het_snp_q\n",
       "53                 anvil_cmg_broad_brain_engle_wes   sequencing                    library-2_pct_exc_dupe\n",
       "54                 anvil_cmg_broad_brain_engle_wes   sequencing             library-2_percent_duplication\n",
       "55                 anvil_cmg_broad_brain_engle_wes   sequencing                  pct_usable_bases_on_bait\n",
       "56                 anvil_cmg_broad_brain_engle_wes   sequencing                       on_bait_vs_selected\n",
       "57                 anvil_cmg_broad_brain_engle_wes   sequencing                           pct_exc_overlap\n",
       "58                 anvil_cmg_broad_brain_engle_wes   sequencing                              pct_off_bait\n",
       "59                 anvil_cmg_broad_brain_engle_wes   sequencing                                 cram_path\n",
       "60                 anvil_cmg_broad_brain_engle_wes   sequencing                            library-1_name\n",
       "61                 anvil_cmg_broad_brain_engle_wes   sequencing                       width_of_99_percent\n",
       "62                 anvil_cmg_broad_brain_engle_wes   sequencing                    median_target_coverage\n",
       "63                 anvil_cmg_broad_brain_engle_wes   sequencing                       width_of_10_percent\n",
       "64                 anvil_cmg_broad_brain_engle_wes   sequencing                             sequencing_id\n",
       "65                 anvil_cmg_broad_brain_engle_wes   sequencing                           near_bait_bases\n",
       "66                 anvil_cmg_broad_brain_engle_wes   sequencing                          pf_hq_error_rate\n",
       "67                 anvil_cmg_broad_brain_engle_wes   sequencing                                 crai_path\n",
       "68                 anvil_cmg_broad_brain_engle_wes   sequencing                       read_pairs_examined\n",
       "69                 anvil_cmg_broad_brain_engle_wes   sequencing                             pct_exc_baseq\n",
       "70                 anvil_cmg_broad_brain_engle_wes   sequencing                       het_snp_sensitivity\n",
       "71                 anvil_cmg_broad_brain_engle_wes   sequencing                              pct_exc_mapq\n",
       "72                 anvil_cmg_broad_brain_engle_wes   sequencing                           fold_enrichment\n",
       "73                 anvil_cmg_broad_brain_engle_wes   sequencing                           on_target_bases\n",
       "74                 anvil_cmg_broad_brain_engle_wes   sequencing                      library-1_read_pairs\n",
       "75                 anvil_cmg_broad_brain_engle_wes   sequencing                      library-2_read_pairs\n",
       "76                 anvil_cmg_broad_brain_engle_wes   sequencing                          mean_read_length\n",
       "77                 anvil_cmg_broad_brain_engle_wes   sequencing                pct_usable_bases_on_target\n",
       "78                 anvil_cmg_broad_brain_engle_wes   sequencing                     pct_target_bases_100x\n",
       "79                 anvil_cmg_broad_brain_engle_wes   sequencing                            strand_balance\n",
       "80                 anvil_cmg_broad_brain_engle_wes   sequencing          library-2_estimated_library_size\n",
       "81                 anvil_cmg_broad_brain_engle_wes   sequencing                        mean_bait_coverage\n",
       "82                 anvil_cmg_broad_brain_engle_wes   sequencing                              pct_exc_dupe\n",
       "83                 anvil_cmg_broad_brain_engle_wes   sequencing                      read_pair_duplicates\n",
       "84                 anvil_cmg_broad_brain_engle_wes   sequencing                                   version\n",
       "85                 anvil_cmg_broad_brain_engle_wes   sequencing                             on_bait_bases\n",
       "86                 anvil_cmg_broad_brain_engle_wes   sequencing                      pct_pf_reads_aligned\n",
       "87                 anvil_cmg_broad_brain_engle_wes   sequencing                          target_territory\n",
       "88                 anvil_cmg_broad_brain_engle_wes   sequencing                          mean_insert_size\n",
       "89                 anvil_cmg_broad_brain_engle_wes   sequencing                            bait_territory\n",
       "90                 anvil_cmg_broad_brain_engle_wes   sequencing                              pct_chimeras\n",
       "91                 anvil_cmg_broad_brain_engle_wes   sequencing                       width_of_50_percent\n",
       "92                 anvil_cmg_broad_brain_engle_wes   sequencing                            library-2_name\n",
       "93                 anvil_cmg_broad_brain_engle_wes   sequencing                library-2_pair_orientation\n",
       "94                 anvil_cmg_broad_brain_engle_wes   sequencing                         pct_contamination\n",
       "95                 anvil_cmg_broad_brain_engle_wes   sequencing                      zero_cvg_targets_pct\n",
       "96                 anvil_cmg_broad_brain_engle_wes   sequencing                library-2_mean_insert_size\n",
       "97                 anvil_cmg_broad_brain_engle_wes   sequencing                library-1_mean_insert_size\n",
       "98                 anvil_cmg_broad_brain_engle_wes   sequencing               collaborator_participant_id\n",
       "99                 anvil_cmg_broad_brain_engle_wes       sample                              02-sample_id\n",
       "100                anvil_cmg_broad_brain_engle_wes       sample                      04-sequencing_center\n",
       "101                anvil_cmg_broad_brain_engle_wes       sample                        03-dbgap_sample_id\n",
       "102                anvil_cmg_broad_brain_engle_wes       sample                                 sample_id\n",
       "103                anvil_cmg_broad_brain_engle_wes       sample                 06-tissue_affected_status\n",
       "104                anvil_cmg_broad_brain_engle_wes       sample                          05-sample_source\n",
       "105                anvil_cmg_broad_brain_engle_wes       sample                             01-subject_id\n",
       "106                anvil_cmg_broad_brain_engle_wes       family                   03-consanguinity_detail\n",
       "107                anvil_cmg_broad_brain_engle_wes       family                           07-family_onset\n",
       "108                anvil_cmg_broad_brain_engle_wes       family                                 family_id\n",
       "109                anvil_cmg_broad_brain_engle_wes       family                              01-family_id\n",
       "110                anvil_cmg_broad_brain_engle_wes       family                         04-pedigree_image\n",
       "111                anvil_cmg_broad_brain_engle_wes       family                        05-pedigree_detail\n",
       "112                anvil_cmg_broad_brain_engle_wes       family                         06-family_history\n",
       "113                anvil_cmg_broad_brain_engle_wes       family                          02-consanguinity\n",
       "114                anvil_cmg_broad_brain_engle_wes    discovery                                16-sv_type\n",
       "115                anvil_cmg_broad_brain_engle_wes    discovery                        18-discovery_notes\n",
       "116                anvil_cmg_broad_brain_engle_wes    discovery                05-inheritance_description\n",
       "117                anvil_cmg_broad_brain_engle_wes    discovery                             01-subject_id\n",
       "118                anvil_cmg_broad_brain_engle_wes    discovery                           17-significance\n",
       "119                anvil_cmg_broad_brain_engle_wes    discovery                                    10-ref\n",
       "120                anvil_cmg_broad_brain_engle_wes    discovery                             14-transcript\n",
       "121                anvil_cmg_broad_brain_engle_wes    discovery                               06-zygosity\n",
       "122                anvil_cmg_broad_brain_engle_wes    discovery                             04-gene_class\n",
       "123                anvil_cmg_broad_brain_engle_wes    discovery                                    11-alt\n",
       "124                anvil_cmg_broad_brain_engle_wes    discovery                   07-variant_genome_build\n",
       "125                anvil_cmg_broad_brain_engle_wes    discovery                                   03-gene\n",
       "126                anvil_cmg_broad_brain_engle_wes    discovery                              discovery_id\n",
       "127                anvil_cmg_broad_brain_engle_wes    discovery                                  08-chrom\n",
       "128                anvil_cmg_broad_brain_engle_wes    discovery                              02-sample_id\n",
       "129                anvil_cmg_broad_brain_engle_wes    discovery                                  13-hgvsp\n",
       "130                anvil_cmg_broad_brain_engle_wes    discovery                                  12-hgvsc\n",
       "131                anvil_cmg_broad_brain_engle_wes    discovery                                    09-pos\n",
       "132                anvil_cmg_broad_brain_engle_wes    discovery                                15-sv_name\n",
       "133     anvil_cmg_broad_genitourinary_sinclair_wes      subject                    19-disease_description\n",
       "134     anvil_cmg_broad_genitourinary_sinclair_wes      subject                             01-subject_id\n",
       "135     anvil_cmg_broad_genitourinary_sinclair_wes      subject                                subject_id\n",
       "136     anvil_cmg_broad_genitourinary_sinclair_wes      subject                           22-age_of_onset\n",
       "137     anvil_cmg_broad_genitourinary_sinclair_wes      subject                      07-multiple_datasets\n",
       "138     anvil_cmg_broad_genitourinary_sinclair_wes      subject                        20-affected_status\n",
       "139     anvil_cmg_broad_genitourinary_sinclair_wes      subject                                11-twin_id\n",
       "140     anvil_cmg_broad_genitourinary_sinclair_wes      subject                               14-ancestry\n",
       "141     anvil_cmg_broad_genitourinary_sinclair_wes      subject                          02-prior_testing\n",
       "142     anvil_cmg_broad_genitourinary_sinclair_wes      subject                   12-proband_relationship\n",
       "143     anvil_cmg_broad_genitourinary_sinclair_wes      subject                  25-phenotype_description\n",
       "144     anvil_cmg_broad_genitourinary_sinclair_wes      subject                                04-pmid_id\n",
       "145     anvil_cmg_broad_genitourinary_sinclair_wes      subject                             03-project_id\n",
       "146     anvil_cmg_broad_genitourinary_sinclair_wes      subject                       06-dbgap_subject_id\n",
       "147     anvil_cmg_broad_genitourinary_sinclair_wes      subject                                    13-sex\n",
       "148     anvil_cmg_broad_genitourinary_sinclair_wes      subject                            23-hpo_present\n",
       "149     anvil_cmg_broad_genitourinary_sinclair_wes      subject                      21-congenital_status\n",
       "150     anvil_cmg_broad_genitourinary_sinclair_wes      subject                              08-family_id\n",
       "151     anvil_cmg_broad_genitourinary_sinclair_wes      subject                            10-maternal_id\n",
       "152     anvil_cmg_broad_genitourinary_sinclair_wes      subject                            09-paternal_id\n",
       "153     anvil_cmg_broad_genitourinary_sinclair_wes      subject                             24-hpo_absent\n",
       "154     anvil_cmg_broad_genitourinary_sinclair_wes      subject                        17-phenotype_group\n",
       "155     anvil_cmg_broad_genitourinary_sinclair_wes      subject                             18-disease_id\n",
       "156     anvil_cmg_broad_genitourinary_sinclair_wes      subject                            26-solve_state\n",
       "157     anvil_cmg_broad_genitourinary_sinclair_wes      subject                16-age_at_last_observation\n",
       "158     anvil_cmg_broad_genitourinary_sinclair_wes      subject                         05-dbgap_study_id\n",
       "159     anvil_cmg_broad_genitourinary_sinclair_wes      subject                        15-ancestry_detail\n",
       "160     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                library-1_pair_orientation\n",
       "161     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                                   project\n",
       "162     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                               total_reads\n",
       "163     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                                  md5_path\n",
       "164     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                       pct_target_bases_2x\n",
       "165     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing          library-1_estimated_library_size\n",
       "166     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                      pct_target_bases_10x\n",
       "167     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                                 data_type\n",
       "168     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                      pct_target_bases_30x\n",
       "169     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                      fold_80_base_penalty\n",
       "170     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                    library-1_pct_exc_dupe\n",
       "171     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                      pct_target_bases_20x\n",
       "172     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                   pf_reads_improper_pairs\n",
       "173     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing              read_pair_optical_duplicates\n",
       "174     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                      mean_target_coverage\n",
       "175     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                    collaborator_sample_id\n",
       "176     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                            off_bait_bases\n",
       "177     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                      pct_target_bases_50x\n",
       "178     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                        median_insert_size\n",
       "179     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing             library-1_percent_duplication\n",
       "180     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                                  bait_set\n",
       "181     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                pct_reads_aligned_in_pairs\n",
       "182     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                        pct_selected_bases\n",
       "183     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                        pct_exc_off_target\n",
       "184     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                                 het_snp_q\n",
       "185     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                  pct_usable_bases_on_bait\n",
       "186     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                       on_bait_vs_selected\n",
       "187     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                           pct_exc_overlap\n",
       "188     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                              pct_off_bait\n",
       "189     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                                 cram_path\n",
       "190     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                            library-1_name\n",
       "191     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                       width_of_99_percent\n",
       "192     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                    median_target_coverage\n",
       "193     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                       width_of_10_percent\n",
       "194     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                             sequencing_id\n",
       "195     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                           near_bait_bases\n",
       "196     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                          pf_hq_error_rate\n",
       "197     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                                 crai_path\n",
       "198     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                       read_pairs_examined\n",
       "199     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                             pct_exc_baseq\n",
       "200     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                       het_snp_sensitivity\n",
       "201     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                              pct_exc_mapq\n",
       "202     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                           fold_enrichment\n",
       "203     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                           on_target_bases\n",
       "204     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                      library-1_read_pairs\n",
       "205     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                          mean_read_length\n",
       "206     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                pct_usable_bases_on_target\n",
       "207     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                            strand_balance\n",
       "208     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                     pct_target_bases_100x\n",
       "209     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                        mean_bait_coverage\n",
       "210     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                              pct_exc_dupe\n",
       "211     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                      read_pair_duplicates\n",
       "212     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                                   version\n",
       "213     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                             on_bait_bases\n",
       "214     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                      pct_pf_reads_aligned\n",
       "215     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                          target_territory\n",
       "216     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                          mean_insert_size\n",
       "217     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                            bait_territory\n",
       "218     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                              pct_chimeras\n",
       "219     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                       width_of_50_percent\n",
       "220     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                         pct_contamination\n",
       "221     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                      zero_cvg_targets_pct\n",
       "222     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                                    sample\n",
       "223     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing                library-1_mean_insert_size\n",
       "224     anvil_cmg_broad_genitourinary_sinclair_wes   sequencing               collaborator_participant_id\n",
       "225     anvil_cmg_broad_genitourinary_sinclair_wes       sample                              02-sample_id\n",
       "226     anvil_cmg_broad_genitourinary_sinclair_wes       sample                      04-sequencing_center\n",
       "227     anvil_cmg_broad_genitourinary_sinclair_wes       sample                        03-dbgap_sample_id\n",
       "228     anvil_cmg_broad_genitourinary_sinclair_wes       sample                                 sample_id\n",
       "229     anvil_cmg_broad_genitourinary_sinclair_wes       sample                 06-tissue_affected_status\n",
       "230     anvil_cmg_broad_genitourinary_sinclair_wes       sample                          05-sample_source\n",
       "231     anvil_cmg_broad_genitourinary_sinclair_wes       sample                             01-subject_id\n",
       "232     anvil_cmg_broad_genitourinary_sinclair_wes       family                   03-consanguinity_detail\n",
       "233     anvil_cmg_broad_genitourinary_sinclair_wes       family                           07-family_onset\n",
       "234     anvil_cmg_broad_genitourinary_sinclair_wes       family                                 family_id\n",
       "235     anvil_cmg_broad_genitourinary_sinclair_wes       family                              01-family_id\n",
       "236     anvil_cmg_broad_genitourinary_sinclair_wes       family                         04-pedigree_image\n",
       "237     anvil_cmg_broad_genitourinary_sinclair_wes       family                        05-pedigree_detail\n",
       "238     anvil_cmg_broad_genitourinary_sinclair_wes       family                         06-family_history\n",
       "239     anvil_cmg_broad_genitourinary_sinclair_wes       family                          02-consanguinity\n",
       "240     anvil_cmg_broad_genitourinary_sinclair_wes    discovery                                16-sv_type\n",
       "241     anvil_cmg_broad_genitourinary_sinclair_wes    discovery                        18-discovery_notes\n",
       "242     anvil_cmg_broad_genitourinary_sinclair_wes    discovery                05-inheritance_description\n",
       "243     anvil_cmg_broad_genitourinary_sinclair_wes    discovery                             01-subject_id\n",
       "244     anvil_cmg_broad_genitourinary_sinclair_wes    discovery                           17-significance\n",
       "245     anvil_cmg_broad_genitourinary_sinclair_wes    discovery                                    10-ref\n",
       "246     anvil_cmg_broad_genitourinary_sinclair_wes    discovery                             14-transcript\n",
       "247     anvil_cmg_broad_genitourinary_sinclair_wes    discovery                               06-zygosity\n",
       "248     anvil_cmg_broad_genitourinary_sinclair_wes    discovery                             04-gene_class\n",
       "249     anvil_cmg_broad_genitourinary_sinclair_wes    discovery                                    11-alt\n",
       "250     anvil_cmg_broad_genitourinary_sinclair_wes    discovery                   07-variant_genome_build\n",
       "251     anvil_cmg_broad_genitourinary_sinclair_wes    discovery                                   03-gene\n",
       "252     anvil_cmg_broad_genitourinary_sinclair_wes    discovery                              discovery_id\n",
       "253     anvil_cmg_broad_genitourinary_sinclair_wes    discovery                                  08-chrom\n",
       "254     anvil_cmg_broad_genitourinary_sinclair_wes    discovery                              02-sample_id\n",
       "255     anvil_cmg_broad_genitourinary_sinclair_wes    discovery                                  13-hgvsp\n",
       "256     anvil_cmg_broad_genitourinary_sinclair_wes    discovery                                  12-hgvsc\n",
       "257     anvil_cmg_broad_genitourinary_sinclair_wes    discovery                                    09-pos\n",
       "258     anvil_cmg_broad_genitourinary_sinclair_wes    discovery                                15-sv_name\n",
       "259            anvil_cmg_broad_orphan_jueppner_wes       family                   03-consanguinity_detail\n",
       "260            anvil_cmg_broad_orphan_jueppner_wes       family                           07-family_onset\n",
       "261            anvil_cmg_broad_orphan_jueppner_wes       family                                 family_id\n",
       "262            anvil_cmg_broad_orphan_jueppner_wes       family                              01-family_id\n",
       "263            anvil_cmg_broad_orphan_jueppner_wes       family                         04-pedigree_image\n",
       "264            anvil_cmg_broad_orphan_jueppner_wes       family                        05-pedigree_detail\n",
       "265            anvil_cmg_broad_orphan_jueppner_wes       family                         06-family_history\n",
       "266            anvil_cmg_broad_orphan_jueppner_wes       family                          02-consanguinity\n",
       "267            anvil_cmg_broad_orphan_jueppner_wes       sample                              02-sample_id\n",
       "268            anvil_cmg_broad_orphan_jueppner_wes       sample                      04-sequencing_center\n",
       "269            anvil_cmg_broad_orphan_jueppner_wes       sample                        03-dbgap_sample_id\n",
       "270            anvil_cmg_broad_orphan_jueppner_wes       sample                                 sample_id\n",
       "271            anvil_cmg_broad_orphan_jueppner_wes       sample                 06-tissue_affected_status\n",
       "272            anvil_cmg_broad_orphan_jueppner_wes       sample                          05-sample_source\n",
       "273            anvil_cmg_broad_orphan_jueppner_wes       sample                             01-subject_id\n",
       "274            anvil_cmg_broad_orphan_jueppner_wes   sequencing                library-1_pair_orientation\n",
       "275            anvil_cmg_broad_orphan_jueppner_wes   sequencing                                   project\n",
       "276            anvil_cmg_broad_orphan_jueppner_wes   sequencing                               total_reads\n",
       "277            anvil_cmg_broad_orphan_jueppner_wes   sequencing                                  md5_path\n",
       "278            anvil_cmg_broad_orphan_jueppner_wes   sequencing                       pct_target_bases_2x\n",
       "279            anvil_cmg_broad_orphan_jueppner_wes   sequencing          library-1_estimated_library_size\n",
       "280            anvil_cmg_broad_orphan_jueppner_wes   sequencing                      pct_target_bases_10x\n",
       "281            anvil_cmg_broad_orphan_jueppner_wes   sequencing                                 data_type\n",
       "282            anvil_cmg_broad_orphan_jueppner_wes   sequencing                      pct_target_bases_30x\n",
       "283            anvil_cmg_broad_orphan_jueppner_wes   sequencing                      fold_80_base_penalty\n",
       "284            anvil_cmg_broad_orphan_jueppner_wes   sequencing                    library-1_pct_exc_dupe\n",
       "285            anvil_cmg_broad_orphan_jueppner_wes   sequencing                      pct_target_bases_20x\n",
       "286            anvil_cmg_broad_orphan_jueppner_wes   sequencing                   pf_reads_improper_pairs\n",
       "287            anvil_cmg_broad_orphan_jueppner_wes   sequencing              read_pair_optical_duplicates\n",
       "288            anvil_cmg_broad_orphan_jueppner_wes   sequencing                      mean_target_coverage\n",
       "289            anvil_cmg_broad_orphan_jueppner_wes   sequencing                    collaborator_sample_id\n",
       "290            anvil_cmg_broad_orphan_jueppner_wes   sequencing                            off_bait_bases\n",
       "291            anvil_cmg_broad_orphan_jueppner_wes   sequencing                      pct_target_bases_50x\n",
       "292            anvil_cmg_broad_orphan_jueppner_wes   sequencing                        median_insert_size\n",
       "293            anvil_cmg_broad_orphan_jueppner_wes   sequencing             library-1_percent_duplication\n",
       "294            anvil_cmg_broad_orphan_jueppner_wes   sequencing                                  bait_set\n",
       "295            anvil_cmg_broad_orphan_jueppner_wes   sequencing                pct_reads_aligned_in_pairs\n",
       "296            anvil_cmg_broad_orphan_jueppner_wes   sequencing                        pct_selected_bases\n",
       "297            anvil_cmg_broad_orphan_jueppner_wes   sequencing                        pct_exc_off_target\n",
       "298            anvil_cmg_broad_orphan_jueppner_wes   sequencing                                 het_snp_q\n",
       "299            anvil_cmg_broad_orphan_jueppner_wes   sequencing                  pct_usable_bases_on_bait\n",
       "300            anvil_cmg_broad_orphan_jueppner_wes   sequencing                       on_bait_vs_selected\n",
       "301            anvil_cmg_broad_orphan_jueppner_wes   sequencing                           pct_exc_overlap\n",
       "302            anvil_cmg_broad_orphan_jueppner_wes   sequencing                              pct_off_bait\n",
       "303            anvil_cmg_broad_orphan_jueppner_wes   sequencing                                 cram_path\n",
       "304            anvil_cmg_broad_orphan_jueppner_wes   sequencing                            library-1_name\n",
       "305            anvil_cmg_broad_orphan_jueppner_wes   sequencing                       width_of_99_percent\n",
       "306            anvil_cmg_broad_orphan_jueppner_wes   sequencing                    median_target_coverage\n",
       "307            anvil_cmg_broad_orphan_jueppner_wes   sequencing                       width_of_10_percent\n",
       "308            anvil_cmg_broad_orphan_jueppner_wes   sequencing                             sequencing_id\n",
       "309            anvil_cmg_broad_orphan_jueppner_wes   sequencing                           near_bait_bases\n",
       "310            anvil_cmg_broad_orphan_jueppner_wes   sequencing                          pf_hq_error_rate\n",
       "311            anvil_cmg_broad_orphan_jueppner_wes   sequencing                                 crai_path\n",
       "312            anvil_cmg_broad_orphan_jueppner_wes   sequencing                       read_pairs_examined\n",
       "313            anvil_cmg_broad_orphan_jueppner_wes   sequencing                             pct_exc_baseq\n",
       "314            anvil_cmg_broad_orphan_jueppner_wes   sequencing                       het_snp_sensitivity\n",
       "315            anvil_cmg_broad_orphan_jueppner_wes   sequencing                              pct_exc_mapq\n",
       "316            anvil_cmg_broad_orphan_jueppner_wes   sequencing                           fold_enrichment\n",
       "317            anvil_cmg_broad_orphan_jueppner_wes   sequencing                           on_target_bases\n",
       "318            anvil_cmg_broad_orphan_jueppner_wes   sequencing                      library-1_read_pairs\n",
       "319            anvil_cmg_broad_orphan_jueppner_wes   sequencing                          mean_read_length\n",
       "320            anvil_cmg_broad_orphan_jueppner_wes   sequencing                pct_usable_bases_on_target\n",
       "321            anvil_cmg_broad_orphan_jueppner_wes   sequencing                            strand_balance\n",
       "322            anvil_cmg_broad_orphan_jueppner_wes   sequencing                     pct_target_bases_100x\n",
       "323            anvil_cmg_broad_orphan_jueppner_wes   sequencing                        mean_bait_coverage\n",
       "324            anvil_cmg_broad_orphan_jueppner_wes   sequencing                              pct_exc_dupe\n",
       "325            anvil_cmg_broad_orphan_jueppner_wes   sequencing                      read_pair_duplicates\n",
       "326            anvil_cmg_broad_orphan_jueppner_wes   sequencing                                   version\n",
       "327            anvil_cmg_broad_orphan_jueppner_wes   sequencing                             on_bait_bases\n",
       "328            anvil_cmg_broad_orphan_jueppner_wes   sequencing                      pct_pf_reads_aligned\n",
       "329            anvil_cmg_broad_orphan_jueppner_wes   sequencing                          target_territory\n",
       "330            anvil_cmg_broad_orphan_jueppner_wes   sequencing                          mean_insert_size\n",
       "331            anvil_cmg_broad_orphan_jueppner_wes   sequencing                            bait_territory\n",
       "332            anvil_cmg_broad_orphan_jueppner_wes   sequencing                              pct_chimeras\n",
       "333            anvil_cmg_broad_orphan_jueppner_wes   sequencing                       width_of_50_percent\n",
       "334            anvil_cmg_broad_orphan_jueppner_wes   sequencing                         pct_contamination\n",
       "335            anvil_cmg_broad_orphan_jueppner_wes   sequencing                      zero_cvg_targets_pct\n",
       "336            anvil_cmg_broad_orphan_jueppner_wes   sequencing                                    sample\n",
       "337            anvil_cmg_broad_orphan_jueppner_wes   sequencing                library-1_mean_insert_size\n",
       "338            anvil_cmg_broad_orphan_jueppner_wes   sequencing               collaborator_participant_id\n",
       "339            anvil_cmg_broad_orphan_jueppner_wes      subject                    19-disease_description\n",
       "340            anvil_cmg_broad_orphan_jueppner_wes      subject                             01-subject_id\n",
       "341            anvil_cmg_broad_orphan_jueppner_wes      subject                                subject_id\n",
       "342            anvil_cmg_broad_orphan_jueppner_wes      subject                           22-age_of_onset\n",
       "343            anvil_cmg_broad_orphan_jueppner_wes      subject                      07-multiple_datasets\n",
       "344            anvil_cmg_broad_orphan_jueppner_wes      subject                        20-affected_status\n",
       "345            anvil_cmg_broad_orphan_jueppner_wes      subject                                11-twin_id\n",
       "346            anvil_cmg_broad_orphan_jueppner_wes      subject                               14-ancestry\n",
       "347            anvil_cmg_broad_orphan_jueppner_wes      subject                          02-prior_testing\n",
       "348            anvil_cmg_broad_orphan_jueppner_wes      subject                   12-proband_relationship\n",
       "349            anvil_cmg_broad_orphan_jueppner_wes      subject                  25-phenotype_description\n",
       "350            anvil_cmg_broad_orphan_jueppner_wes      subject                                04-pmid_id\n",
       "351            anvil_cmg_broad_orphan_jueppner_wes      subject                             03-project_id\n",
       "352            anvil_cmg_broad_orphan_jueppner_wes      subject                       06-dbgap_subject_id\n",
       "353            anvil_cmg_broad_orphan_jueppner_wes      subject                                    13-sex\n",
       "354            anvil_cmg_broad_orphan_jueppner_wes      subject                            23-hpo_present\n",
       "355            anvil_cmg_broad_orphan_jueppner_wes      subject                      21-congenital_status\n",
       "356            anvil_cmg_broad_orphan_jueppner_wes      subject                              08-family_id\n",
       "357            anvil_cmg_broad_orphan_jueppner_wes      subject                            10-maternal_id\n",
       "358            anvil_cmg_broad_orphan_jueppner_wes      subject                            09-paternal_id\n",
       "359            anvil_cmg_broad_orphan_jueppner_wes      subject                             24-hpo_absent\n",
       "360            anvil_cmg_broad_orphan_jueppner_wes      subject                        17-phenotype_group\n",
       "361            anvil_cmg_broad_orphan_jueppner_wes      subject                             18-disease_id\n",
       "362            anvil_cmg_broad_orphan_jueppner_wes      subject                            26-solve_state\n",
       "363            anvil_cmg_broad_orphan_jueppner_wes      subject                16-age_at_last_observation\n",
       "364            anvil_cmg_broad_orphan_jueppner_wes      subject                         05-dbgap_study_id\n",
       "365            anvil_cmg_broad_orphan_jueppner_wes      subject                        15-ancestry_detail\n",
       "366   anvil_cmg_broad_genitourinary_hirschhorn_wes  participant               collaborator_participant_id\n",
       "367   anvil_cmg_broad_genitourinary_hirschhorn_wes  participant                                    gender\n",
       "368   anvil_cmg_broad_genitourinary_hirschhorn_wes  participant                            participant_id\n",
       "369   anvil_cmg_broad_genitourinary_hirschhorn_wes      subject                    19-disease_description\n",
       "370   anvil_cmg_broad_genitourinary_hirschhorn_wes      subject                             01-subject_id\n",
       "371   anvil_cmg_broad_genitourinary_hirschhorn_wes      subject                                subject_id\n",
       "372   anvil_cmg_broad_genitourinary_hirschhorn_wes      subject                           22-age_of_onset\n",
       "373   anvil_cmg_broad_genitourinary_hirschhorn_wes      subject                      07-multiple_datasets\n",
       "374   anvil_cmg_broad_genitourinary_hirschhorn_wes      subject                        20-affected_status\n",
       "375   anvil_cmg_broad_genitourinary_hirschhorn_wes      subject                                11-twin_id\n",
       "376   anvil_cmg_broad_genitourinary_hirschhorn_wes      subject                               14-ancestry\n",
       "377   anvil_cmg_broad_genitourinary_hirschhorn_wes      subject                          02-prior_testing\n",
       "378   anvil_cmg_broad_genitourinary_hirschhorn_wes      subject                   12-proband_relationship\n",
       "379   anvil_cmg_broad_genitourinary_hirschhorn_wes      subject                  25-phenotype_description\n",
       "380   anvil_cmg_broad_genitourinary_hirschhorn_wes      subject                                04-pmid_id\n",
       "381   anvil_cmg_broad_genitourinary_hirschhorn_wes      subject                             03-project_id\n",
       "382   anvil_cmg_broad_genitourinary_hirschhorn_wes      subject                       06-dbgap_subject_id\n",
       "383   anvil_cmg_broad_genitourinary_hirschhorn_wes      subject                                    13-sex\n",
       "384   anvil_cmg_broad_genitourinary_hirschhorn_wes      subject                            23-hpo_present\n",
       "385   anvil_cmg_broad_genitourinary_hirschhorn_wes      subject                      21-congenital_status\n",
       "386   anvil_cmg_broad_genitourinary_hirschhorn_wes      subject                              08-family_id\n",
       "387   anvil_cmg_broad_genitourinary_hirschhorn_wes      subject                            10-maternal_id\n",
       "388   anvil_cmg_broad_genitourinary_hirschhorn_wes      subject                            09-paternal_id\n",
       "389   anvil_cmg_broad_genitourinary_hirschhorn_wes      subject                             24-hpo_absent\n",
       "390   anvil_cmg_broad_genitourinary_hirschhorn_wes      subject                        17-phenotype_group\n",
       "391   anvil_cmg_broad_genitourinary_hirschhorn_wes      subject                             18-disease_id\n",
       "392   anvil_cmg_broad_genitourinary_hirschhorn_wes      subject                            26-solve_state\n",
       "393   anvil_cmg_broad_genitourinary_hirschhorn_wes      subject                16-age_at_last_observation\n",
       "394   anvil_cmg_broad_genitourinary_hirschhorn_wes      subject                         05-dbgap_study_id\n",
       "395   anvil_cmg_broad_genitourinary_hirschhorn_wes      subject                        15-ancestry_detail\n",
       "396   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                    hs_het_snp_sensitivity\n",
       "397   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                                at_dropout\n",
       "398   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing         library-1_unpaired_reads_examined\n",
       "399   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                                is_primary\n",
       "400   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                   hs_pct_target_bases_20x\n",
       "401   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                         hs_off_bait_bases\n",
       "402   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                                novel_titv\n",
       "403   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                num_in_db_snp_multiallelic\n",
       "404   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                    rna_pct_intronic_bases\n",
       "405   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                      wgs_pct_exc_unpaired\n",
       "406   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                      pct_target_bases_10x\n",
       "407   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                       rna_ribosomal_bases\n",
       "408   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                       hs_pf_bases_aligned\n",
       "409   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                              novel_indels\n",
       "410   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                      mean_target_coverage\n",
       "411   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                   pf_hq_median_mismatches\n",
       "412   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                            off_bait_bases\n",
       "413   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                                hom_as_het\n",
       "414   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing        rna_pct_r2_transcript_strand_reads\n",
       "415   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing        rna_pct_r1_transcript_strand_reads\n",
       "416   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                             pf_indel_rate\n",
       "417   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                pct_reads_aligned_in_pairs\n",
       "418   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                        pct_selected_bases\n",
       "419   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                              rna_pf_bases\n",
       "420   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                       hs_target_territory\n",
       "421   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing              num_in_db_snp_complex_indels\n",
       "422   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                 median_absolute_deviation\n",
       "423   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing            library-1_read_pair_duplicates\n",
       "424   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                rna_incorrect_strand_reads\n",
       "425   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                      total_complex_indels\n",
       "426   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                               wgs_pct_50x\n",
       "427   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                       width_of_90_percent\n",
       "428   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                               wgs_pct_20x\n",
       "429   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                           hs_pct_pf_reads\n",
       "430   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                         hs_hs_penalty_20x\n",
       "431   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                             rna_utr_bases\n",
       "432   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                        hs_pct_pf_uq_reads\n",
       "433   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                    hs_on_bait_vs_selected\n",
       "434   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                         rna_ignored_reads\n",
       "435   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                               wgs_pct_25x\n",
       "436   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                 hs_bait_design_efficiency\n",
       "437   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                       het_snp_sensitivity\n",
       "438   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                              pct_exc_mapq\n",
       "439   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                           fold_enrichment\n",
       "440   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                          mean_read_length\n",
       "441   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                       width_of_30_percent\n",
       "442   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                               gc_nc_40_59\n",
       "443   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                 hs_median_target_coverage\n",
       "444   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                       width_of_40_percent\n",
       "445   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                              pct_exc_dupe\n",
       "446   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                              gc_nc_80_100\n",
       "447   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                          rna_coding_bases\n",
       "448   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                   rna_pct_ribosomal_bases\n",
       "449   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                            bait_territory\n",
       "450   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                          hom_as_other_hom\n",
       "451   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                             filtered_snps\n",
       "452   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                    hs_pct_target_bases_2x\n",
       "453   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                        hs_pf_unique_reads\n",
       "454   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                library-1_mean_insert_size\n",
       "455   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                          pf_hq_error_rate\n",
       "456   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                               window_size\n",
       "457   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                library-1_pair_orientation\n",
       "458   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                  rna_correct_strand_reads\n",
       "459   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                        rna_pct_mrna_bases\n",
       "460   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing               hs_pct_usable_bases_on_bait\n",
       "461   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                               wgs_pct_70x\n",
       "462   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                  library-2_unmapped_reads\n",
       "463   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                       width_of_80_percent\n",
       "464   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing          library-1_estimated_library_size\n",
       "465   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                      pct_target_bases_30x\n",
       "466   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                              total_indels\n",
       "467   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                               wgs_pct_30x\n",
       "468   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                                dbsnp_titv\n",
       "469   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                         hs_hs_penalty_30x\n",
       "470   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                    collaborator_sample_id\n",
       "471   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                                read_pairs\n",
       "472   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                      rna_pf_aligned_bases\n",
       "473   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                                wgs_pct_5x\n",
       "474   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                                  category\n",
       "475   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                                  bait_set\n",
       "476   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                         rna_pct_utr_bases\n",
       "477   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                               wgs_pct_15x\n",
       "478   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                         wgs_pct_exc_baseq\n",
       "479   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                             hs_gc_dropout\n",
       "480   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing             library-2_percent_duplication\n",
       "481   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                   wgs_het_snp_sensitivity\n",
       "482   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                                  num_snps\n",
       "483   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                           max_insert_size\n",
       "484   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                       width_of_99_percent\n",
       "485   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing            haplotypes_confidently_matchin\n",
       "486   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                    median_target_coverage\n",
       "487   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                       width_of_10_percent\n",
       "488   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                             sequencing_id\n",
       "489   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing  library-2_secondary_or_supplementary_rds\n",
       "490   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                           near_bait_bases\n",
       "491   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                           wgs_sd_coverage\n",
       "492   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                             pct_exc_baseq\n",
       "493   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                   hs_zero_cvg_targets_pct\n",
       "494   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing             library-1_read_pairs_examined\n",
       "495   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                               hs_pf_reads\n",
       "496   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                        hs_pct_exc_overlap\n",
       "497   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                      library-1_read_pairs\n",
       "498   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing        rna_num_r2_transcript_strand_reads\n",
       "499   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                     pct_target_bases_100x\n",
       "500   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                             wgs_het_snp_q\n",
       "501   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                   hs_pct_target_bases_30x\n",
       "502   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                       novel_ins_del_ratio\n",
       "503   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                         hs_hs_penalty_40x\n",
       "504   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                          pf_reads_aligned\n",
       "505   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                      pct_pf_reads_aligned\n",
       "506   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                      rna_intergenic_bases\n",
       "507   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                               gc_nc_20_39\n",
       "508   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                              pct_chimeras\n",
       "509   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing            haplotypes_confidently_checked\n",
       "510   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                   pf_hq_aligned_q20_bases\n",
       "511   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                      zero_cvg_targets_pct\n",
       "512   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                             num_in_db_snp\n",
       "513   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                        hs_near_bait_bases\n",
       "514   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                             hs_at_dropout\n",
       "515   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                               participant\n",
       "516   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                          wgs_pct_exc_dupe\n",
       "517   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                               wgs_pct_10x\n",
       "518   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                   hs_fold_80_base_penalty\n",
       "519   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                               wgs_pct_60x\n",
       "520   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                                  pf_reads\n",
       "521   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                          wgs_mad_coverage\n",
       "522   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                                  md5_path\n",
       "523   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing    library-1_read_pair_optical_duplicates\n",
       "524   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                         wgs_pct_exc_total\n",
       "525   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                   hs_pct_target_bases_50x\n",
       "526   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                       pf_hq_aligned_bases\n",
       "527   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                 rna_num_unexplained_reads\n",
       "528   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                                       pdo\n",
       "529   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                    library-1_pct_exc_dupe\n",
       "530   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                                het_as_hom\n",
       "531   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                            hs_genome_size\n",
       "532   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                   hs_mean_target_coverage\n",
       "533   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                          crai_or_bai_path\n",
       "534   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                                mean_depth\n",
       "535   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                          ll_random_sample\n",
       "536   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                              pct_pf_reads\n",
       "537   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                         hs_hs_penalty_10x\n",
       "538   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                      pct_target_bases_20x\n",
       "539   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                      pct_target_bases_50x\n",
       "540   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                              wgs_pct_100x\n",
       "541   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                    reads_aligned_in_pairs\n",
       "542   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing              rna_pct_correct_strand_reads\n",
       "543   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                             aligned_reads\n",
       "544   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                              hs_het_snp_q\n",
       "545   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing             library-1_percent_duplication\n",
       "546   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                          hs_pct_exc_baseq\n",
       "547   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                                 pct_dbsnp\n",
       "548   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                         hs_hs_penalty_50x\n",
       "549   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                            total_clusters\n",
       "550   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                       wgs_median_coverage\n",
       "551   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                    rna_median_cv_coverage\n",
       "552   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                            num_singletons\n",
       "553   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                  pct_usable_bases_on_bait\n",
       "554   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                        snp_reference_bias\n",
       "555   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                       on_bait_vs_selected\n",
       "556   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                      num_in_db_snp_indels\n",
       "557   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing          rna_median_5prime_to_3prime_bias\n",
       "558   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                    hs_min_target_coverage\n",
       "559   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                       lod_expected_sample\n",
       "560   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                                 cram_path\n",
       "561   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                   total_multiallelic_snps\n",
       "562   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                           hs_pct_exc_dupe\n",
       "563   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                       ll_predicted_contam\n",
       "564   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                   reference_sequence_name\n",
       "565   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                              ll_no_contam\n",
       "566   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                            pf_noise_reads\n",
       "567   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                        rna_intronic_bases\n",
       "568   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                                gc_dropout\n",
       "569   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                          pf_aligned_bases\n",
       "570   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing        rna_num_r1_transcript_strand_reads\n",
       "571   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                              release_date\n",
       "572   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                pct_usable_bases_on_target\n",
       "573   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                                 num_reads\n",
       "574   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                       width_of_60_percent\n",
       "575   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                        sequencing_project\n",
       "576   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                      rna_pct_coding_bases\n",
       "577   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                        ll_expected_sample\n",
       "578   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                        mean_bait_coverage\n",
       "579   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                 haplotypes_with_genotypes\n",
       "580   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                                    lc-set\n",
       "581   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                             pct_chimerism\n",
       "582   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                             on_bait_bases\n",
       "583   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                          research_project\n",
       "584   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                    rna_median_5prime_bias\n",
       "585   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                                   version\n",
       "586   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                       wgs_pct_exc_overlap\n",
       "587   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                              sample_alias\n",
       "588   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                         pct_contamination\n",
       "589   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                               read_groups\n",
       "590   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                               pct_adapter\n",
       "591   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                                    sample\n",
       "592   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                    hs_max_target_coverage\n",
       "593   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                                wgs_pct_1x\n",
       "594   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                         hs_bait_territory\n",
       "595   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                   hs_pct_target_bases_10x\n",
       "596   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing               collaborator_participant_id\n",
       "597   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                       pf_hq_aligned_reads\n",
       "598   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                                   project\n",
       "599   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                               total_reads\n",
       "600   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                          wgs_pct_exc_mapq\n",
       "601   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                                total_snps\n",
       "602   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                        hs_on_target_bases\n",
       "603   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                       pct_target_bases_2x\n",
       "604   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                          pf_mismatch_rate\n",
       "605   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                                 data_type\n",
       "606   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                        hs_fold_enrichment\n",
       "607   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                      fold_80_base_penalty\n",
       "608   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                          hs_on_bait_bases\n",
       "609   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                  hs_pct_target_bases_100x\n",
       "610   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                        hs_hs_library_size\n",
       "611   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                      wgs_genome_territory\n",
       "612   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                     hs_mean_bait_coverage\n",
       "613   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                          pct_dbsnp_indels\n",
       "614   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                          pair_orientation\n",
       "615   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing             library-2_read_pairs_examined\n",
       "616   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                        median_insert_size\n",
       "617   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                    hs_pct_target_bases_1x\n",
       "618   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                          individual_alias\n",
       "619   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing        library-1_unpaired_read_duplicates\n",
       "620   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                                read_group\n",
       "621   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                       width_of_20_percent\n",
       "622   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                                gc_nc_0_19\n",
       "623   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                        pct_exc_off_target\n",
       "624   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                               sample_type\n",
       "625   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                                 het_snp_q\n",
       "626   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                hs_pct_pf_uq_reads_aligned\n",
       "627   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                           hs_pct_exc_mapq\n",
       "628   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing             hs_pct_usable_bases_on_target\n",
       "629   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                           pct_exc_overlap\n",
       "630   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                                novel_snps\n",
       "631   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                   hs_pct_target_bases_40x\n",
       "632   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                     hs_pct_exc_off_target\n",
       "633   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                              pct_off_bait\n",
       "634   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                        standard_deviation\n",
       "635   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                               wgs_pct_80x\n",
       "636   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                            library-1_name\n",
       "637   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing        library-2_unpaired_read_duplicates\n",
       "638   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                       dbsnp_ins_del_ratio\n",
       "639   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                  library-1_unmapped_reads\n",
       "640   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                     hs_pct_selected_bases\n",
       "641   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                            hs_total_reads\n",
       "642   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing    library-2_read_pair_optical_duplicates\n",
       "643   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                               wgs_pct_40x\n",
       "644   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                                 crai_path\n",
       "645   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing            library-2_read_pair_duplicates\n",
       "646   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                               wgs_pct_90x\n",
       "647   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                            strand_balance\n",
       "648   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                           on_target_bases\n",
       "649   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                    hs_pf_uq_bases_aligned\n",
       "650   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                         wgs_mean_coverage\n",
       "651   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                        hs_hs_penalty_100x\n",
       "652   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing          library-2_estimated_library_size\n",
       "653   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                                bad_cycles\n",
       "654   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                           min_insert_size\n",
       "655   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                        wgs_pct_exc_capped\n",
       "656   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                       width_of_70_percent\n",
       "657   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing         library-2_unpaired_reads_examined\n",
       "658   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                          target_territory\n",
       "659   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                    hs_pf_uq_reads_aligned\n",
       "660   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                          mean_insert_size\n",
       "661   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                           filtered_indels\n",
       "662   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                               gc_nc_60_79\n",
       "663   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                           hs_pct_off_bait\n",
       "664   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing  library-1_secondary_or_supplementary_rds\n",
       "665   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                            library-2_name\n",
       "666   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                  rna_pct_intergenic_bases\n",
       "667   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                      rna_pct_usable_bases\n",
       "668   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                          cram_or_bam_path\n",
       "669   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                       width_of_50_percent\n",
       "670   anvil_cmg_broad_genitourinary_hirschhorn_wes   sequencing                    rna_median_3prime_bias\n",
       "671   anvil_cmg_broad_genitourinary_hirschhorn_wes       sample                              02-sample_id\n",
       "672   anvil_cmg_broad_genitourinary_hirschhorn_wes       sample                      04-sequencing_center\n",
       "673   anvil_cmg_broad_genitourinary_hirschhorn_wes       sample                        03-dbgap_sample_id\n",
       "674   anvil_cmg_broad_genitourinary_hirschhorn_wes       sample                                 sample_id\n",
       "675   anvil_cmg_broad_genitourinary_hirschhorn_wes       sample                 06-tissue_affected_status\n",
       "676   anvil_cmg_broad_genitourinary_hirschhorn_wes       sample                          05-sample_source\n",
       "677   anvil_cmg_broad_genitourinary_hirschhorn_wes       sample                             01-subject_id\n",
       "678   anvil_cmg_broad_genitourinary_hirschhorn_wes       family                   03-consanguinity_detail\n",
       "679   anvil_cmg_broad_genitourinary_hirschhorn_wes       family                           07-family_onset\n",
       "680   anvil_cmg_broad_genitourinary_hirschhorn_wes       family                                 family_id\n",
       "681   anvil_cmg_broad_genitourinary_hirschhorn_wes       family                              01-family_id\n",
       "682   anvil_cmg_broad_genitourinary_hirschhorn_wes       family                         04-pedigree_image\n",
       "683   anvil_cmg_broad_genitourinary_hirschhorn_wes       family                        05-pedigree_detail\n",
       "684   anvil_cmg_broad_genitourinary_hirschhorn_wes       family                         06-family_history\n",
       "685   anvil_cmg_broad_genitourinary_hirschhorn_wes       family                          02-consanguinity\n",
       "686   anvil_cmg_broad_genitourinary_hirschhorn_wes    discovery                                16-sv_type\n",
       "687   anvil_cmg_broad_genitourinary_hirschhorn_wes    discovery                        18-discovery_notes\n",
       "688   anvil_cmg_broad_genitourinary_hirschhorn_wes    discovery                05-inheritance_description\n",
       "689   anvil_cmg_broad_genitourinary_hirschhorn_wes    discovery                             01-subject_id\n",
       "690   anvil_cmg_broad_genitourinary_hirschhorn_wes    discovery                           17-significance\n",
       "691   anvil_cmg_broad_genitourinary_hirschhorn_wes    discovery                                    10-ref\n",
       "692   anvil_cmg_broad_genitourinary_hirschhorn_wes    discovery                             14-transcript\n",
       "693   anvil_cmg_broad_genitourinary_hirschhorn_wes    discovery                               06-zygosity\n",
       "694   anvil_cmg_broad_genitourinary_hirschhorn_wes    discovery                             04-gene_class\n",
       "695   anvil_cmg_broad_genitourinary_hirschhorn_wes    discovery                                    11-alt\n",
       "696   anvil_cmg_broad_genitourinary_hirschhorn_wes    discovery                   07-variant_genome_build\n",
       "697   anvil_cmg_broad_genitourinary_hirschhorn_wes    discovery                                   03-gene\n",
       "698   anvil_cmg_broad_genitourinary_hirschhorn_wes    discovery                              discovery_id\n",
       "699   anvil_cmg_broad_genitourinary_hirschhorn_wes    discovery                                  08-chrom\n",
       "700   anvil_cmg_broad_genitourinary_hirschhorn_wes    discovery                              02-sample_id\n",
       "701   anvil_cmg_broad_genitourinary_hirschhorn_wes    discovery                                  13-hgvsp\n",
       "702   anvil_cmg_broad_genitourinary_hirschhorn_wes    discovery                                  12-hgvsc\n",
       "703   anvil_cmg_broad_genitourinary_hirschhorn_wes    discovery                                    09-pos\n",
       "704   anvil_cmg_broad_genitourinary_hirschhorn_wes    discovery                                15-sv_name\n",
       "705               anvil_cmg_broad_orphan_scott_wes      subject                    19-disease_description\n",
       "706               anvil_cmg_broad_orphan_scott_wes      subject                             01-subject_id\n",
       "707               anvil_cmg_broad_orphan_scott_wes      subject                                subject_id\n",
       "708               anvil_cmg_broad_orphan_scott_wes      subject                           22-age_of_onset\n",
       "709               anvil_cmg_broad_orphan_scott_wes      subject                      07-multiple_datasets\n",
       "710               anvil_cmg_broad_orphan_scott_wes      subject                        20-affected_status\n",
       "711               anvil_cmg_broad_orphan_scott_wes      subject                                11-twin_id\n",
       "712               anvil_cmg_broad_orphan_scott_wes      subject                               14-ancestry\n",
       "713               anvil_cmg_broad_orphan_scott_wes      subject                          02-prior_testing\n",
       "714               anvil_cmg_broad_orphan_scott_wes      subject                   12-proband_relationship\n",
       "715               anvil_cmg_broad_orphan_scott_wes      subject                  25-phenotype_description\n",
       "716               anvil_cmg_broad_orphan_scott_wes      subject                                04-pmid_id\n",
       "717               anvil_cmg_broad_orphan_scott_wes      subject                             03-project_id\n",
       "718               anvil_cmg_broad_orphan_scott_wes      subject                       06-dbgap_subject_id\n",
       "719               anvil_cmg_broad_orphan_scott_wes      subject                                    13-sex\n",
       "720               anvil_cmg_broad_orphan_scott_wes      subject                            23-hpo_present\n",
       "721               anvil_cmg_broad_orphan_scott_wes      subject                      21-congenital_status\n",
       "722               anvil_cmg_broad_orphan_scott_wes      subject                              08-family_id\n",
       "723               anvil_cmg_broad_orphan_scott_wes      subject                            10-maternal_id\n",
       "724               anvil_cmg_broad_orphan_scott_wes      subject                            09-paternal_id\n",
       "725               anvil_cmg_broad_orphan_scott_wes      subject                             24-hpo_absent\n",
       "726               anvil_cmg_broad_orphan_scott_wes      subject                        17-phenotype_group\n",
       "727               anvil_cmg_broad_orphan_scott_wes      subject                             18-disease_id\n",
       "728               anvil_cmg_broad_orphan_scott_wes      subject                            26-solve_state\n",
       "729               anvil_cmg_broad_orphan_scott_wes      subject                16-age_at_last_observation\n",
       "730               anvil_cmg_broad_orphan_scott_wes      subject                         05-dbgap_study_id\n",
       "731               anvil_cmg_broad_orphan_scott_wes      subject                        15-ancestry_detail\n",
       "732               anvil_cmg_broad_orphan_scott_wes   sequencing                library-1_pair_orientation\n",
       "733               anvil_cmg_broad_orphan_scott_wes   sequencing                                   project\n",
       "734               anvil_cmg_broad_orphan_scott_wes   sequencing                               total_reads\n",
       "735               anvil_cmg_broad_orphan_scott_wes   sequencing                                  md5_path\n",
       "736               anvil_cmg_broad_orphan_scott_wes   sequencing                       pct_target_bases_2x\n",
       "737               anvil_cmg_broad_orphan_scott_wes   sequencing          library-1_estimated_library_size\n",
       "738               anvil_cmg_broad_orphan_scott_wes   sequencing                      pct_target_bases_10x\n",
       "739               anvil_cmg_broad_orphan_scott_wes   sequencing                                 data_type\n",
       "740               anvil_cmg_broad_orphan_scott_wes   sequencing                      pct_target_bases_30x\n",
       "741               anvil_cmg_broad_orphan_scott_wes   sequencing                      fold_80_base_penalty\n",
       "742               anvil_cmg_broad_orphan_scott_wes   sequencing                    library-1_pct_exc_dupe\n",
       "743               anvil_cmg_broad_orphan_scott_wes   sequencing                      pct_target_bases_20x\n",
       "744               anvil_cmg_broad_orphan_scott_wes   sequencing                   pf_reads_improper_pairs\n",
       "745               anvil_cmg_broad_orphan_scott_wes   sequencing              read_pair_optical_duplicates\n",
       "746               anvil_cmg_broad_orphan_scott_wes   sequencing                      mean_target_coverage\n",
       "747               anvil_cmg_broad_orphan_scott_wes   sequencing                    collaborator_sample_id\n",
       "748               anvil_cmg_broad_orphan_scott_wes   sequencing                            off_bait_bases\n",
       "749               anvil_cmg_broad_orphan_scott_wes   sequencing                      pct_target_bases_50x\n",
       "750               anvil_cmg_broad_orphan_scott_wes   sequencing                        median_insert_size\n",
       "751               anvil_cmg_broad_orphan_scott_wes   sequencing             library-1_percent_duplication\n",
       "752               anvil_cmg_broad_orphan_scott_wes   sequencing                                  bait_set\n",
       "753               anvil_cmg_broad_orphan_scott_wes   sequencing                pct_reads_aligned_in_pairs\n",
       "754               anvil_cmg_broad_orphan_scott_wes   sequencing                        pct_selected_bases\n",
       "755               anvil_cmg_broad_orphan_scott_wes   sequencing                        pct_exc_off_target\n",
       "756               anvil_cmg_broad_orphan_scott_wes   sequencing                                    sample\n",
       "757               anvil_cmg_broad_orphan_scott_wes   sequencing                                 het_snp_q\n",
       "758               anvil_cmg_broad_orphan_scott_wes   sequencing                    library-2_pct_exc_dupe\n",
       "759               anvil_cmg_broad_orphan_scott_wes   sequencing             library-2_percent_duplication\n",
       "760               anvil_cmg_broad_orphan_scott_wes   sequencing                  pct_usable_bases_on_bait\n",
       "761               anvil_cmg_broad_orphan_scott_wes   sequencing                       on_bait_vs_selected\n",
       "762               anvil_cmg_broad_orphan_scott_wes   sequencing                           pct_exc_overlap\n",
       "763               anvil_cmg_broad_orphan_scott_wes   sequencing                              pct_off_bait\n",
       "764               anvil_cmg_broad_orphan_scott_wes   sequencing                                 cram_path\n",
       "765               anvil_cmg_broad_orphan_scott_wes   sequencing                            library-1_name\n",
       "766               anvil_cmg_broad_orphan_scott_wes   sequencing                       width_of_99_percent\n",
       "767               anvil_cmg_broad_orphan_scott_wes   sequencing                    median_target_coverage\n",
       "768               anvil_cmg_broad_orphan_scott_wes   sequencing                       width_of_10_percent\n",
       "769               anvil_cmg_broad_orphan_scott_wes   sequencing                             sequencing_id\n",
       "770               anvil_cmg_broad_orphan_scott_wes   sequencing                           near_bait_bases\n",
       "771               anvil_cmg_broad_orphan_scott_wes   sequencing                          pf_hq_error_rate\n",
       "772               anvil_cmg_broad_orphan_scott_wes   sequencing                                 crai_path\n",
       "773               anvil_cmg_broad_orphan_scott_wes   sequencing                       read_pairs_examined\n",
       "774               anvil_cmg_broad_orphan_scott_wes   sequencing                             pct_exc_baseq\n",
       "775               anvil_cmg_broad_orphan_scott_wes   sequencing                       het_snp_sensitivity\n",
       "776               anvil_cmg_broad_orphan_scott_wes   sequencing                              pct_exc_mapq\n",
       "777               anvil_cmg_broad_orphan_scott_wes   sequencing                           fold_enrichment\n",
       "778               anvil_cmg_broad_orphan_scott_wes   sequencing                           on_target_bases\n",
       "779               anvil_cmg_broad_orphan_scott_wes   sequencing                      library-1_read_pairs\n",
       "780               anvil_cmg_broad_orphan_scott_wes   sequencing                      library-2_read_pairs\n",
       "781               anvil_cmg_broad_orphan_scott_wes   sequencing                          mean_read_length\n",
       "782               anvil_cmg_broad_orphan_scott_wes   sequencing                pct_usable_bases_on_target\n",
       "783               anvil_cmg_broad_orphan_scott_wes   sequencing                     pct_target_bases_100x\n",
       "784               anvil_cmg_broad_orphan_scott_wes   sequencing                            strand_balance\n",
       "785               anvil_cmg_broad_orphan_scott_wes   sequencing          library-2_estimated_library_size\n",
       "786               anvil_cmg_broad_orphan_scott_wes   sequencing                        mean_bait_coverage\n",
       "787               anvil_cmg_broad_orphan_scott_wes   sequencing                              pct_exc_dupe\n",
       "788               anvil_cmg_broad_orphan_scott_wes   sequencing                      read_pair_duplicates\n",
       "789               anvil_cmg_broad_orphan_scott_wes   sequencing                                   version\n",
       "790               anvil_cmg_broad_orphan_scott_wes   sequencing                             on_bait_bases\n",
       "791               anvil_cmg_broad_orphan_scott_wes   sequencing                      pct_pf_reads_aligned\n",
       "792               anvil_cmg_broad_orphan_scott_wes   sequencing                          target_territory\n",
       "793               anvil_cmg_broad_orphan_scott_wes   sequencing                          mean_insert_size\n",
       "794               anvil_cmg_broad_orphan_scott_wes   sequencing                            bait_territory\n",
       "795               anvil_cmg_broad_orphan_scott_wes   sequencing                              pct_chimeras\n",
       "796               anvil_cmg_broad_orphan_scott_wes   sequencing                       width_of_50_percent\n",
       "797               anvil_cmg_broad_orphan_scott_wes   sequencing                            library-2_name\n",
       "798               anvil_cmg_broad_orphan_scott_wes   sequencing                library-2_pair_orientation\n",
       "799               anvil_cmg_broad_orphan_scott_wes   sequencing                         pct_contamination\n",
       "800               anvil_cmg_broad_orphan_scott_wes   sequencing                      zero_cvg_targets_pct\n",
       "801               anvil_cmg_broad_orphan_scott_wes   sequencing                library-2_mean_insert_size\n",
       "802               anvil_cmg_broad_orphan_scott_wes   sequencing                library-1_mean_insert_size\n",
       "803               anvil_cmg_broad_orphan_scott_wes   sequencing               collaborator_participant_id\n",
       "804               anvil_cmg_broad_orphan_scott_wes       sample                              02-sample_id\n",
       "805               anvil_cmg_broad_orphan_scott_wes       sample                      04-sequencing_center\n",
       "806               anvil_cmg_broad_orphan_scott_wes       sample                        03-dbgap_sample_id\n",
       "807               anvil_cmg_broad_orphan_scott_wes       sample                                 sample_id\n",
       "808               anvil_cmg_broad_orphan_scott_wes       sample                 06-tissue_affected_status\n",
       "809               anvil_cmg_broad_orphan_scott_wes       sample                          05-sample_source\n",
       "810               anvil_cmg_broad_orphan_scott_wes       sample                             01-subject_id\n",
       "811               anvil_cmg_broad_orphan_scott_wes       family                   03-consanguinity_detail\n",
       "812               anvil_cmg_broad_orphan_scott_wes       family                           07-family_onset\n",
       "813               anvil_cmg_broad_orphan_scott_wes       family                                 family_id\n",
       "814               anvil_cmg_broad_orphan_scott_wes       family                              01-family_id\n",
       "815               anvil_cmg_broad_orphan_scott_wes       family                         04-pedigree_image\n",
       "816               anvil_cmg_broad_orphan_scott_wes       family                        05-pedigree_detail\n",
       "817               anvil_cmg_broad_orphan_scott_wes       family                         06-family_history\n",
       "818               anvil_cmg_broad_orphan_scott_wes       family                          02-consanguinity\n",
       "819               anvil_cmg_broad_orphan_scott_wes    discovery                                16-sv_type\n",
       "820               anvil_cmg_broad_orphan_scott_wes    discovery                        18-discovery_notes\n",
       "821               anvil_cmg_broad_orphan_scott_wes    discovery                05-inheritance_description\n",
       "822               anvil_cmg_broad_orphan_scott_wes    discovery                             01-subject_id\n",
       "823               anvil_cmg_broad_orphan_scott_wes    discovery                           17-significance\n",
       "824               anvil_cmg_broad_orphan_scott_wes    discovery                                    10-ref\n",
       "825               anvil_cmg_broad_orphan_scott_wes    discovery                             14-transcript\n",
       "826               anvil_cmg_broad_orphan_scott_wes    discovery                               06-zygosity\n",
       "827               anvil_cmg_broad_orphan_scott_wes    discovery                             04-gene_class\n",
       "828               anvil_cmg_broad_orphan_scott_wes    discovery                                    11-alt\n",
       "829               anvil_cmg_broad_orphan_scott_wes    discovery                   07-variant_genome_build\n",
       "830               anvil_cmg_broad_orphan_scott_wes    discovery                                   03-gene\n",
       "831               anvil_cmg_broad_orphan_scott_wes    discovery                              discovery_id\n",
       "832               anvil_cmg_broad_orphan_scott_wes    discovery                                  08-chrom\n",
       "833               anvil_cmg_broad_orphan_scott_wes    discovery                              02-sample_id\n",
       "834               anvil_cmg_broad_orphan_scott_wes    discovery                                  13-hgvsp\n",
       "835               anvil_cmg_broad_orphan_scott_wes    discovery                                  12-hgvsc\n",
       "836               anvil_cmg_broad_orphan_scott_wes    discovery                                    09-pos\n",
       "837               anvil_cmg_broad_orphan_scott_wes    discovery                                15-sv_name\n",
       "838          anvil_cmg_broad_orphan_vcgs-white_wes  participant               collaborator_participant_id\n",
       "839          anvil_cmg_broad_orphan_vcgs-white_wes  participant                                    gender\n",
       "840          anvil_cmg_broad_orphan_vcgs-white_wes  participant                            participant_id\n",
       "841          anvil_cmg_broad_orphan_vcgs-white_wes      subject                    19-disease_description\n",
       "842          anvil_cmg_broad_orphan_vcgs-white_wes      subject                             01-subject_id\n",
       "843          anvil_cmg_broad_orphan_vcgs-white_wes      subject                                subject_id\n",
       "844          anvil_cmg_broad_orphan_vcgs-white_wes      subject                           22-age_of_onset\n",
       "845          anvil_cmg_broad_orphan_vcgs-white_wes      subject                      07-multiple_datasets\n",
       "846          anvil_cmg_broad_orphan_vcgs-white_wes      subject                        20-affected_status\n",
       "847          anvil_cmg_broad_orphan_vcgs-white_wes      subject                                11-twin_id\n",
       "848          anvil_cmg_broad_orphan_vcgs-white_wes      subject                               14-ancestry\n",
       "849          anvil_cmg_broad_orphan_vcgs-white_wes      subject                          02-prior_testing\n",
       "850          anvil_cmg_broad_orphan_vcgs-white_wes      subject                   12-proband_relationship\n",
       "851          anvil_cmg_broad_orphan_vcgs-white_wes      subject                  25-phenotype_description\n",
       "852          anvil_cmg_broad_orphan_vcgs-white_wes      subject                                04-pmid_id\n",
       "853          anvil_cmg_broad_orphan_vcgs-white_wes      subject                             03-project_id\n",
       "854          anvil_cmg_broad_orphan_vcgs-white_wes      subject                       06-dbgap_subject_id\n",
       "855          anvil_cmg_broad_orphan_vcgs-white_wes      subject                                    13-sex\n",
       "856          anvil_cmg_broad_orphan_vcgs-white_wes      subject                            23-hpo_present\n",
       "857          anvil_cmg_broad_orphan_vcgs-white_wes      subject                      21-congenital_status\n",
       "858          anvil_cmg_broad_orphan_vcgs-white_wes      subject                              08-family_id\n",
       "859          anvil_cmg_broad_orphan_vcgs-white_wes      subject                            10-maternal_id\n",
       "860          anvil_cmg_broad_orphan_vcgs-white_wes      subject                            09-paternal_id\n",
       "861          anvil_cmg_broad_orphan_vcgs-white_wes      subject                             24-hpo_absent\n",
       "862          anvil_cmg_broad_orphan_vcgs-white_wes      subject                        17-phenotype_group\n",
       "863          anvil_cmg_broad_orphan_vcgs-white_wes      subject                             18-disease_id\n",
       "864          anvil_cmg_broad_orphan_vcgs-white_wes      subject                            26-solve_state\n",
       "865          anvil_cmg_broad_orphan_vcgs-white_wes      subject                16-age_at_last_observation\n",
       "866          anvil_cmg_broad_orphan_vcgs-white_wes      subject                         05-dbgap_study_id\n",
       "867          anvil_cmg_broad_orphan_vcgs-white_wes      subject                        15-ancestry_detail\n",
       "868          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                      pct_target_bases_10x\n",
       "869          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                                  20x_rate\n",
       "870          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                      mean_target_coverage\n",
       "871          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                            off_bait_bases\n",
       "872          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                        pct_selected_bases\n",
       "873          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                    library-2_pct_exc_dupe\n",
       "874          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                                       pdo\n",
       "875          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                         exc_unpaired_rate\n",
       "876          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                             pf_reads_rate\n",
       "877          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                              adapter_rate\n",
       "878          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                       het_snp_sensitivity\n",
       "879          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                           fold_enrichment\n",
       "880          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                              pct_exc_mapq\n",
       "881          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                          mean_read_length\n",
       "882          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                              pct_exc_dupe\n",
       "883          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                            bait_territory\n",
       "884          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                library-2_mean_insert_size\n",
       "885          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                library-1_mean_insert_size\n",
       "886          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                library-1_pair_orientation\n",
       "887          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing          library-1_estimated_library_size\n",
       "888          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                      pct_target_bases_30x\n",
       "889          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                          genome_territory\n",
       "890          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                    collaborator_sample_id\n",
       "891          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                                  bait_set\n",
       "892          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing             library-2_percent_duplication\n",
       "893          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                    median_target_coverage\n",
       "894          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                             sequencing_id\n",
       "895          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                           near_bait_bases\n",
       "896          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                             pct_exc_baseq\n",
       "897          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                      library-1_read_pairs\n",
       "898          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                     pct_target_bases_100x\n",
       "899          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                          pf_reads_aligned\n",
       "900          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing               reads_aligned_in_pairs_rate\n",
       "901          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                              pct_chimeras\n",
       "902          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                   pf_hq_aligned_q20_bases\n",
       "903          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                      zero_cvg_targets_pct\n",
       "904          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                                  pf_reads\n",
       "905          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                             mean_coverage\n",
       "906          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                                  md5_path\n",
       "907          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                            exc_baseq_rate\n",
       "908          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                       pf_hq_aligned_bases\n",
       "909          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                    library-1_pct_exc_dupe\n",
       "910          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                      pct_target_bases_20x\n",
       "911          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                      pct_target_bases_50x\n",
       "912          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                    reads_aligned_in_pairs\n",
       "913          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing             library-1_percent_duplication\n",
       "914          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                                  10x_rate\n",
       "915          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                  pct_usable_bases_on_bait\n",
       "916          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                          exc_overlap_rate\n",
       "917          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                       on_bait_vs_selected\n",
       "918          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                                 cram_path\n",
       "919          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                   reference_sequence_name\n",
       "920          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                            pf_noise_reads\n",
       "921          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                          pf_aligned_bases\n",
       "922          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                              release_date\n",
       "923          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                pct_usable_bases_on_target\n",
       "924          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                        mean_bait_coverage\n",
       "925          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                                   version\n",
       "926          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                             on_bait_bases\n",
       "927          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                                    sample\n",
       "928          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing               collaborator_participant_id\n",
       "929          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                        contamination_rate\n",
       "930          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                       pf_hq_aligned_reads\n",
       "931          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                                   project\n",
       "932          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                               total_reads\n",
       "933          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                             exc_dupe_rate\n",
       "934          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                       pct_target_bases_2x\n",
       "935          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                          pf_mismatch_rate\n",
       "936          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                                 data_type\n",
       "937          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                      fold_80_base_penalty\n",
       "938          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                            exc_total_rate\n",
       "939          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                        pct_exc_off_target\n",
       "940          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                                 het_snp_q\n",
       "941          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                     pf_reads_aligned_rate\n",
       "942          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                           pct_exc_overlap\n",
       "943          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                              pct_off_bait\n",
       "944          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                            library-1_name\n",
       "945          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                                 crai_path\n",
       "946          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                           on_target_bases\n",
       "947          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                              chimera_rate\n",
       "948          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing          library-2_estimated_library_size\n",
       "949          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                             exc_mapq_rate\n",
       "950          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                          target_territory\n",
       "951          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                            root_sample_id\n",
       "952          anvil_cmg_broad_orphan_vcgs-white_wes   sequencing                            library-2_name\n",
       "953          anvil_cmg_broad_orphan_vcgs-white_wes       sample                              02-sample_id\n",
       "954          anvil_cmg_broad_orphan_vcgs-white_wes       sample                      04-sequencing_center\n",
       "955          anvil_cmg_broad_orphan_vcgs-white_wes       sample                        03-dbgap_sample_id\n",
       "956          anvil_cmg_broad_orphan_vcgs-white_wes       sample                                 sample_id\n",
       "957          anvil_cmg_broad_orphan_vcgs-white_wes       sample                 06-tissue_affected_status\n",
       "958          anvil_cmg_broad_orphan_vcgs-white_wes       sample                          05-sample_source\n",
       "959          anvil_cmg_broad_orphan_vcgs-white_wes       sample                             01-subject_id\n",
       "960          anvil_cmg_broad_orphan_vcgs-white_wes       family                   03-consanguinity_detail\n",
       "961          anvil_cmg_broad_orphan_vcgs-white_wes       family                           07-family_onset\n",
       "962          anvil_cmg_broad_orphan_vcgs-white_wes       family                                 family_id\n",
       "963          anvil_cmg_broad_orphan_vcgs-white_wes       family                              01-family_id\n",
       "964          anvil_cmg_broad_orphan_vcgs-white_wes       family                         04-pedigree_image\n",
       "965          anvil_cmg_broad_orphan_vcgs-white_wes       family                        05-pedigree_detail\n",
       "966          anvil_cmg_broad_orphan_vcgs-white_wes       family                         06-family_history\n",
       "967          anvil_cmg_broad_orphan_vcgs-white_wes       family                          02-consanguinity\n",
       "968          anvil_cmg_broad_orphan_vcgs-white_wes    discovery                                16-sv_type\n",
       "969          anvil_cmg_broad_orphan_vcgs-white_wes    discovery                        18-discovery_notes\n",
       "970          anvil_cmg_broad_orphan_vcgs-white_wes    discovery                05-inheritance_description\n",
       "971          anvil_cmg_broad_orphan_vcgs-white_wes    discovery                             01-subject_id\n",
       "972          anvil_cmg_broad_orphan_vcgs-white_wes    discovery                           17-significance\n",
       "973          anvil_cmg_broad_orphan_vcgs-white_wes    discovery                                    10-ref\n",
       "974          anvil_cmg_broad_orphan_vcgs-white_wes    discovery                             14-transcript\n",
       "975          anvil_cmg_broad_orphan_vcgs-white_wes    discovery                               06-zygosity\n",
       "976          anvil_cmg_broad_orphan_vcgs-white_wes    discovery                             04-gene_class\n",
       "977          anvil_cmg_broad_orphan_vcgs-white_wes    discovery                                    11-alt\n",
       "978          anvil_cmg_broad_orphan_vcgs-white_wes    discovery                   07-variant_genome_build\n",
       "979          anvil_cmg_broad_orphan_vcgs-white_wes    discovery                                   03-gene\n",
       "980          anvil_cmg_broad_orphan_vcgs-white_wes    discovery                              discovery_id\n",
       "981          anvil_cmg_broad_orphan_vcgs-white_wes    discovery                                  08-chrom\n",
       "982          anvil_cmg_broad_orphan_vcgs-white_wes    discovery                              02-sample_id\n",
       "983          anvil_cmg_broad_orphan_vcgs-white_wes    discovery                                  13-hgvsp\n",
       "984          anvil_cmg_broad_orphan_vcgs-white_wes    discovery                                  12-hgvsc\n",
       "985          anvil_cmg_broad_orphan_vcgs-white_wes    discovery                                    09-pos\n",
       "986          anvil_cmg_broad_orphan_vcgs-white_wes    discovery                                15-sv_name\n",
       "987    anvil_cmg_broad_stillbirth_wilkins-haug_wes      subject                    19-disease_description\n",
       "988    anvil_cmg_broad_stillbirth_wilkins-haug_wes      subject                             01-subject_id\n",
       "989    anvil_cmg_broad_stillbirth_wilkins-haug_wes      subject                                subject_id\n",
       "990    anvil_cmg_broad_stillbirth_wilkins-haug_wes      subject                           22-age_of_onset\n",
       "991    anvil_cmg_broad_stillbirth_wilkins-haug_wes      subject                      07-multiple_datasets\n",
       "992    anvil_cmg_broad_stillbirth_wilkins-haug_wes      subject                        20-affected_status\n",
       "993    anvil_cmg_broad_stillbirth_wilkins-haug_wes      subject                                11-twin_id\n",
       "994    anvil_cmg_broad_stillbirth_wilkins-haug_wes      subject                               14-ancestry\n",
       "995    anvil_cmg_broad_stillbirth_wilkins-haug_wes      subject                          02-prior_testing\n",
       "996    anvil_cmg_broad_stillbirth_wilkins-haug_wes      subject                   12-proband_relationship\n",
       "997    anvil_cmg_broad_stillbirth_wilkins-haug_wes      subject                  25-phenotype_description\n",
       "998    anvil_cmg_broad_stillbirth_wilkins-haug_wes      subject                                04-pmid_id\n",
       "999    anvil_cmg_broad_stillbirth_wilkins-haug_wes      subject                             03-project_id\n",
       "1000   anvil_cmg_broad_stillbirth_wilkins-haug_wes      subject                       06-dbgap_subject_id\n",
       "1001   anvil_cmg_broad_stillbirth_wilkins-haug_wes      subject                                    13-sex\n",
       "1002   anvil_cmg_broad_stillbirth_wilkins-haug_wes      subject                            23-hpo_present\n",
       "1003   anvil_cmg_broad_stillbirth_wilkins-haug_wes      subject                      21-congenital_status\n",
       "1004   anvil_cmg_broad_stillbirth_wilkins-haug_wes      subject                              08-family_id\n",
       "1005   anvil_cmg_broad_stillbirth_wilkins-haug_wes      subject                            10-maternal_id\n",
       "1006   anvil_cmg_broad_stillbirth_wilkins-haug_wes      subject                            09-paternal_id\n",
       "1007   anvil_cmg_broad_stillbirth_wilkins-haug_wes      subject                             24-hpo_absent\n",
       "1008   anvil_cmg_broad_stillbirth_wilkins-haug_wes      subject                        17-phenotype_group\n",
       "1009   anvil_cmg_broad_stillbirth_wilkins-haug_wes      subject                             18-disease_id\n",
       "1010   anvil_cmg_broad_stillbirth_wilkins-haug_wes      subject                            26-solve_state\n",
       "1011   anvil_cmg_broad_stillbirth_wilkins-haug_wes      subject                16-age_at_last_observation\n",
       "1012   anvil_cmg_broad_stillbirth_wilkins-haug_wes      subject                         05-dbgap_study_id\n",
       "1013   anvil_cmg_broad_stillbirth_wilkins-haug_wes      subject                        15-ancestry_detail\n",
       "1014   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                library-1_pair_orientation\n",
       "1015   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                                   project\n",
       "1016   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                               total_reads\n",
       "1017   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                                  md5_path\n",
       "1018   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                       pct_target_bases_2x\n",
       "1019   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing          library-1_estimated_library_size\n",
       "1020   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                      pct_target_bases_10x\n",
       "1021   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                                 data_type\n",
       "1022   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                      pct_target_bases_30x\n",
       "1023   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                      fold_80_base_penalty\n",
       "1024   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                    library-1_pct_exc_dupe\n",
       "1025   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                      pct_target_bases_20x\n",
       "1026   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                   pf_reads_improper_pairs\n",
       "1027   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing              read_pair_optical_duplicates\n",
       "1028   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                      mean_target_coverage\n",
       "1029   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                    collaborator_sample_id\n",
       "1030   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                            off_bait_bases\n",
       "1031   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                      pct_target_bases_50x\n",
       "1032   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                        median_insert_size\n",
       "1033   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing             library-1_percent_duplication\n",
       "1034   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                                  bait_set\n",
       "1035   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                pct_reads_aligned_in_pairs\n",
       "1036   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                        pct_selected_bases\n",
       "1037   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                        pct_exc_off_target\n",
       "1038   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                                 het_snp_q\n",
       "1039   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                  pct_usable_bases_on_bait\n",
       "1040   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                       on_bait_vs_selected\n",
       "1041   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                           pct_exc_overlap\n",
       "1042   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                              pct_off_bait\n",
       "1043   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                                 cram_path\n",
       "1044   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                            library-1_name\n",
       "1045   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                       width_of_99_percent\n",
       "1046   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                    median_target_coverage\n",
       "1047   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                       width_of_10_percent\n",
       "1048   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                             sequencing_id\n",
       "1049   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                           near_bait_bases\n",
       "1050   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                          pf_hq_error_rate\n",
       "1051   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                                 crai_path\n",
       "1052   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                       read_pairs_examined\n",
       "1053   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                             pct_exc_baseq\n",
       "1054   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                       het_snp_sensitivity\n",
       "1055   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                              pct_exc_mapq\n",
       "1056   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                           fold_enrichment\n",
       "1057   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                           on_target_bases\n",
       "1058   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                      library-1_read_pairs\n",
       "1059   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                          mean_read_length\n",
       "1060   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                pct_usable_bases_on_target\n",
       "1061   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                            strand_balance\n",
       "1062   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                     pct_target_bases_100x\n",
       "1063   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                        mean_bait_coverage\n",
       "1064   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                              pct_exc_dupe\n",
       "1065   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                      read_pair_duplicates\n",
       "1066   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                                   version\n",
       "1067   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                             on_bait_bases\n",
       "1068   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                      pct_pf_reads_aligned\n",
       "1069   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                          target_territory\n",
       "1070   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                          mean_insert_size\n",
       "1071   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                            bait_territory\n",
       "1072   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                              pct_chimeras\n",
       "1073   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                       width_of_50_percent\n",
       "1074   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                         pct_contamination\n",
       "1075   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                      zero_cvg_targets_pct\n",
       "1076   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                                    sample\n",
       "1077   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing                library-1_mean_insert_size\n",
       "1078   anvil_cmg_broad_stillbirth_wilkins-haug_wes   sequencing               collaborator_participant_id\n",
       "1079   anvil_cmg_broad_stillbirth_wilkins-haug_wes       sample                              02-sample_id\n",
       "1080   anvil_cmg_broad_stillbirth_wilkins-haug_wes       sample                      04-sequencing_center\n",
       "1081   anvil_cmg_broad_stillbirth_wilkins-haug_wes       sample                        03-dbgap_sample_id\n",
       "1082   anvil_cmg_broad_stillbirth_wilkins-haug_wes       sample                                 sample_id\n",
       "1083   anvil_cmg_broad_stillbirth_wilkins-haug_wes       sample                 06-tissue_affected_status\n",
       "1084   anvil_cmg_broad_stillbirth_wilkins-haug_wes       sample                          05-sample_source\n",
       "1085   anvil_cmg_broad_stillbirth_wilkins-haug_wes       sample                             01-subject_id\n",
       "1086   anvil_cmg_broad_stillbirth_wilkins-haug_wes       family                   03-consanguinity_detail\n",
       "1087   anvil_cmg_broad_stillbirth_wilkins-haug_wes       family                           07-family_onset\n",
       "1088   anvil_cmg_broad_stillbirth_wilkins-haug_wes       family                                 family_id\n",
       "1089   anvil_cmg_broad_stillbirth_wilkins-haug_wes       family                              01-family_id\n",
       "1090   anvil_cmg_broad_stillbirth_wilkins-haug_wes       family                         04-pedigree_image\n",
       "1091   anvil_cmg_broad_stillbirth_wilkins-haug_wes       family                        05-pedigree_detail\n",
       "1092   anvil_cmg_broad_stillbirth_wilkins-haug_wes       family                         06-family_history\n",
       "1093   anvil_cmg_broad_stillbirth_wilkins-haug_wes       family                          02-consanguinity\n",
       "1094   anvil_cmg_broad_stillbirth_wilkins-haug_wes    discovery                                16-sv_type\n",
       "1095   anvil_cmg_broad_stillbirth_wilkins-haug_wes    discovery                        18-discovery_notes\n",
       "1096   anvil_cmg_broad_stillbirth_wilkins-haug_wes    discovery                05-inheritance_description\n",
       "1097   anvil_cmg_broad_stillbirth_wilkins-haug_wes    discovery                             01-subject_id\n",
       "1098   anvil_cmg_broad_stillbirth_wilkins-haug_wes    discovery                           17-significance\n",
       "1099   anvil_cmg_broad_stillbirth_wilkins-haug_wes    discovery                                    10-ref\n",
       "1100   anvil_cmg_broad_stillbirth_wilkins-haug_wes    discovery                             14-transcript\n",
       "1101   anvil_cmg_broad_stillbirth_wilkins-haug_wes    discovery                               06-zygosity\n",
       "1102   anvil_cmg_broad_stillbirth_wilkins-haug_wes    discovery                             04-gene_class\n",
       "1103   anvil_cmg_broad_stillbirth_wilkins-haug_wes    discovery                                    11-alt\n",
       "1104   anvil_cmg_broad_stillbirth_wilkins-haug_wes    discovery                   07-variant_genome_build\n",
       "1105   anvil_cmg_broad_stillbirth_wilkins-haug_wes    discovery                                   03-gene\n",
       "1106   anvil_cmg_broad_stillbirth_wilkins-haug_wes    discovery                              discovery_id\n",
       "1107   anvil_cmg_broad_stillbirth_wilkins-haug_wes    discovery                                  08-chrom\n",
       "1108   anvil_cmg_broad_stillbirth_wilkins-haug_wes    discovery                              02-sample_id\n",
       "1109   anvil_cmg_broad_stillbirth_wilkins-haug_wes    discovery                                  13-hgvsp\n",
       "1110   anvil_cmg_broad_stillbirth_wilkins-haug_wes    discovery                                  12-hgvsc\n",
       "1111   anvil_cmg_broad_stillbirth_wilkins-haug_wes    discovery                                    09-pos\n",
       "1112   anvil_cmg_broad_stillbirth_wilkins-haug_wes    discovery                                15-sv_name"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "## Collecting the schema for a specific workspace\n",
    "from firecloud import api as fapi\n",
    "ws_project = \"anvil-datastorage\"\n",
    "ws_name_list = [\n",
    "'ANVIL_CMG_BROAD_BRAIN_ENGLE_WES',\n",
    "'ANVIL_CMG_Broad_Genitourinary_Sinclair_WES',\n",
    "'ANVIL_CMG_Broad_Orphan_Jueppner_WES',\n",
    "'AnVIL_CMG_Broad_Genitourinary_Hirschhorn_WES',\n",
    "'AnVIL_CMG_Broad_Orphan_Scott_WES',\n",
    "'AnVIL_CMG_Broad_Orphan_VCGS-White_WES',\n",
    "'AnVIL_CMG_Broad_Stillbirth_Wilkins-Haug_WES',\n",
    "]\n",
    "schema_fields = []\n",
    "\n",
    "# Loop through workspaces\n",
    "for ws_name in ws_name_list:\n",
    "\n",
    "    try:\n",
    "        # Collect and record all entity types in workspace\n",
    "        response_etypes = fapi.list_entity_types(ws_project, ws_name)\n",
    "        dict_all_etypes = json.loads(response_etypes.text)\n",
    "        etypes_list = [key for key in dict_all_etypes.keys()]\n",
    "\n",
    "        # Loop through entity types and parse result to build schema\n",
    "        if etypes_list:\n",
    "            for etype in etypes_list:\n",
    "                column_set = set()\n",
    "                column_set.add(dict_all_etypes[etype][\"idName\"])\n",
    "                for attr_key in dict_all_etypes[etype][\"attributeNames\"]:\n",
    "                    column_set.add(attr_key)\n",
    "                for column in column_set:\n",
    "                    column_entry = []\n",
    "                    column_entry = [ws_name.lower(), etype.lower(), column.lower()]\n",
    "                    schema_fields.append(column_entry)\n",
    "    except:\n",
    "        pass\n",
    "\n",
    "# Convert to dataframe and display\n",
    "df = pd.DataFrame(schema_fields, columns = [\"workspace_name\", \"table_name\", \"column_name\"])\n",
    "display(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
